{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create MEGAN tree and GraphIan visualization\n",
    "\n",
    "- create csv to export to MEGAN\n",
    "- create GraphIan layers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import datetime\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "#For illustrator import:\n",
    "plt.rcParams['pdf.fonttype'] = 42\n",
    "plt.rcParams['ps.fonttype'] = 42"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Functions\n",
    "\n",
    "#Raw Read Numbers\n",
    "def make_rawread(infile):\n",
    "    #infile = OTU_table_taxa_all.txt\n",
    "    df = pd.read_csv(infile, sep='\\t')\n",
    "    df.rename(columns={'DUP_ID':'OTU'}, inplace=True)\n",
    "    df.drop('taxonomy', axis=1, inplace=True)\n",
    "    df.drop('repseq', axis=1, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    return df\n",
    "    \n",
    "#metadata handling and sample order\n",
    "def make_metadata(infile):\n",
    "    df = pd.read_csv(infile)\n",
    "    #make all column names with underscore\n",
    "    columns_l = list(df)\n",
    "    columns_l = [c.replace('-', '_') for c in columns_l]\n",
    "    df.columns = columns_l\n",
    "    #extract site name\n",
    "    df['site'] = df['sample_name'].str.extract('([a-zA-Z0-9]+)')\n",
    "    df_full = df[:]\n",
    "    #add in some missing location data\n",
    "    df_full.loc[df_full['site']=='UC5', 'dec_lat'] = 32\n",
    "    df_full.loc[df_full['site']=='UC5', 'dec_long'] = '.118.17'\n",
    "    df_full['dec_long'] = df_full['dec_long'].str.replace('^\\.','-')\n",
    "    df_full.set_index('sample_name', inplace=True)\n",
    "    df= df[['Description', 'Treatment', 'sample_name']]\n",
    "    df['comb'] = df['Description']+'_'+df['Treatment']+'_'+df['sample_name']\n",
    "    df['site_Order']=df['sample_name'].str.replace('GOC2a', '16').str.replace('GOC2b', '17').str.replace('NTC', '0')\n",
    "    df['site_Order']=df['site_Order'].str.extract('(\\d+)')\n",
    "    df.set_index('sample_name', inplace=True)\n",
    "    df['site_Order']= df['site_Order'].astype(int)\n",
    "    df.sort_values('site_Order', ascending=True, inplace=True)\n",
    "    return df_full, df\n",
    "\n",
    "#DESEQ Numbers\n",
    "def make_deseq(infile):\n",
    "    df = pd.read_csv(infile, sep=',')\n",
    "    df.rename(columns={'Unnamed: 0':'OTU'}, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    return df\n",
    "\n",
    "#Rarefied Read Numbers (From R script); make compositional\n",
    "def make_rarereads_comp(infile):\n",
    "    df= pd.read_csv(infile, sep=',')\n",
    "    df.rename(columns={'Unnamed: 0':'OTU'}, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    df=df.T\n",
    "    df['Total']=df.sum(axis=1)\n",
    "    tot_reads = df['Total'].tolist()\n",
    "    tot_reads = set(tot_reads)\n",
    "    print('Rarefied read count:', tot_reads)\n",
    "    tot_reads = df['Total'].tolist()[0]\n",
    "    df = df/tot_reads *100\n",
    "    df.drop('Total', axis=1, inplace=True)\n",
    "    df=df.T\n",
    "    return df\n",
    "\n",
    "def make_rarereads(infile):\n",
    "    df= pd.read_csv(infile, sep=',')\n",
    "    df.rename(columns={'Unnamed: 0':'OTU'}, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    return df\n",
    "\n",
    "#Taxa table\n",
    "def make_taxa(infile):\n",
    "    #infile = OTU_table_taxa_all.txt\n",
    "    df = pd.read_csv(infile, sep='\\t')\n",
    "    df.rename(columns={'DUP_ID':'OTU'}, inplace=True)\n",
    "    df = df[['taxonomy', 'OTU']]\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    df['Kingdom']=df['taxonomy'].str.split(\"\\'\").str[1]\n",
    "    df['Phylum']=df['taxonomy'].str.split(\"\\'\").str[3]\n",
    "    df['Class']=df['taxonomy'].str.split(\"\\'\").str[5]\n",
    "    df['Order']=df['taxonomy'].str.split(\"\\'\").str[7]\n",
    "    df['Family']=df['taxonomy'].str.split(\"\\'\").str[9]\n",
    "    df['Genus']=df['taxonomy'].str.split(\"\\'\").str[11]\n",
    "    df['Species']=df['taxonomy'].str.split(\"\\'\").str[13]\n",
    "    df=df.drop('taxonomy', axis=1)\n",
    "    return df\n",
    "\n",
    "#Filtered Taxa table\n",
    "def make_Ftaxa(infile):\n",
    "    df = pd.read_csv(infile, sep=',')\n",
    "    df.rename(columns={'OTU_ID':'OTU'}, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    df=df[df.columns[-7:]]\n",
    "    return df\n",
    "\n",
    "#extract sequences\n",
    "def make_seq(infile):\n",
    "    #infile = OTU_table_taxa_all.txt\n",
    "    df = pd.read_csv(infile, sep='\\t')\n",
    "    df.rename(columns={'DUP_ID':'OTU'}, inplace=True)\n",
    "    df.set_index('OTU', inplace=True)\n",
    "    df=df[['repseq']]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import Banzai Results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "COI\n",
      "Rarefied read count: {129363}\n",
      "18S\n",
      "Rarefied read count: {28676}\n",
      "/Users/kpitz/Projects/Gulf_of_California/GOC_18S_COI_Combined_Cutadapt/GOC_18SCOI_\n"
     ]
    }
   ],
   "source": [
    "#COI M6\n",
    "print('COI')\n",
    "#OTU_table_taxa_all.txt location\n",
    "file_loc1 = '/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/COI/Analysis_20190921_1416/all_lib/Post_Blast_20190925_0914/OTU_table_taxa_all.txt'\n",
    "#metadata\n",
    "meta_file = '/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/COI/Analysis_20190921_1416/GOC_20190921_1416_COI_analysis_metadata.csv'\n",
    "\n",
    "#DESEQ data\n",
    "#file2 = '/Users/kpitz/Projects/Gulf_of_California/Deseq/COI_GOC_DEseq_122117.csv'\n",
    "#rarefied data\n",
    "file3 = '/Users/kpitz/Projects/MBON/Rarefied_Data_unmerged/GOC_COI_OTU_Table_092619_M6.csv'\n",
    "#Filtered OTU table\n",
    "file4 ='/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/COI/Analysis_20190921_1416/all_lib/Filtered_OTU_taxa_table_all.csv'\n",
    "\n",
    "raw_COI = make_rawread(file_loc1)\n",
    "meta_COI, samp_lim_COI = make_metadata(meta_file)\n",
    "#deseq_COI = make_deseq(file2)\n",
    "rare_COI = make_rarereads(file3)\n",
    "rare_comp_COI = make_rarereads_comp(file3)\n",
    "taxa_COI = make_taxa(file_loc1)\n",
    "Ftaxa_COI = make_Ftaxa(file4)\n",
    "seq_COI = make_seq(file_loc1)\n",
    "\n",
    "#18S_M6\n",
    "print('18S')\n",
    "#OTU_table_taxa_all.txt location\n",
    "file_loc1 = '/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/18S/Analysis_20190924_1129/all_lib/Post_Blast_20190930_1343/OTU_table_taxa_all.txt'\n",
    "#metadata\n",
    "meta_file = '/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/18S/Analysis_20190924_1129/GOC_18S_Metadata.csv'\n",
    "#DESEQ data\n",
    "#file2 = '/Users/kpitz/Projects/Gulf_of_California/Deseq/18S_GOC_DEseq_122117.csv'\n",
    "#rarefied data\n",
    "file3 = '/Users/kpitz/Projects/MBON/Rarefied_Data_unmerged/GOC_18S_OTU_Table_100119_M6.csv'\n",
    "#Filtered OTU table\n",
    "file4 = '/Users/kpitz/Projects/Gulf_of_California/Cutadapt_Results/18S/Analysis_20190924_1129/all_lib/Filtered_OTU_taxa_table_all.csv'\n",
    "\n",
    "\n",
    "raw_18S = make_rawread(file_loc1)\n",
    "meta_18S, samp_lim_18S = make_metadata(meta_file)\n",
    "#deseq_18S = make_deseq(file2)\n",
    "rare_18S = make_rarereads(file3)\n",
    "rare_comp_18S = make_rarereads_comp(file3)\n",
    "taxa_18S = make_taxa(file_loc1)\n",
    "Ftaxa_18S = make_Ftaxa(file4)\n",
    "seq_18S = make_seq(file_loc1)\n",
    "\n",
    "\n",
    "#Directory for saving Figures\n",
    "plot_dir = '/Users/kpitz/Projects/Gulf_of_California/GOC_18S_COI_Combined_Cutadapt/'\n",
    "plot_name = 'GOC_18SCOI_'\n",
    "Plot_str = plot_dir + plot_name\n",
    "print(Plot_str)\n",
    "\n",
    "#Use to change the name of the databases\n",
    "name1 = 'GOC_18SCOI_'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import Filtered Banzai Results\n",
    "\n",
    "Created in GOC_COI_Filter_Data.ipynb\n",
    "\n",
    "- /Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_seq_table_092519.csv\n",
    "- /Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_otu_table_092519.csv\n",
    "- /Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_taxa_table_092519.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['order', 'tag_sequence', 'primer_sequence_F', 'primer_sequence_R', 'library_tag_combo', 'library', 'sample_type', 'locus', 'tag_number', 'R1', 'R2', 'Treatment', 'Time_of_Day', 'Description', 'Description_3', 'site', 'SEQ', 'BOTTLE', 'DEPTH', 'CRUISE', 'PLATFORM', 'DEC_LAT', 'DEC_LONG', 'TMP', 'SAL', 'CHL_GFF', 'PRESSURE', 'NO3', 'OXY_ML', 'RDEP', 'TRANSMISS', 'SIG_T', 'FLUOR', 'DATE_TIME', 'cruise', 'SEQAvg_dg', 'AvgOfTMP', 'StDevOfTMP', 'CountOfTMP', 'AvgOfSAL1', 'StDevOfSAL', 'CountOfSAL', 'AvgOfCHLA', 'StDevOfCHLA', 'CountOfCHLA', 'AvgOfOXY_ML1', 'CountOfOXY_ML1', 'CountOfOXY_ML', 'AvgOfTRANSMISS', 'StDevOfTRANSMISS', 'CountOfTRANSMISS', 'AvgOfSIGMA_THETA', 'StDevOfSIGMA_THETA', 'CountOfSIGMA_THETA']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>order</th>\n",
       "      <th>tag_sequence</th>\n",
       "      <th>primer_sequence_F</th>\n",
       "      <th>primer_sequence_R</th>\n",
       "      <th>library_tag_combo</th>\n",
       "      <th>library</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>locus</th>\n",
       "      <th>tag_number</th>\n",
       "      <th>R1</th>\n",
       "      <th>...</th>\n",
       "      <th>CountOfCHLA</th>\n",
       "      <th>AvgOfOXY_ML1</th>\n",
       "      <th>CountOfOXY_ML1</th>\n",
       "      <th>CountOfOXY_ML</th>\n",
       "      <th>AvgOfTRANSMISS</th>\n",
       "      <th>StDevOfTRANSMISS</th>\n",
       "      <th>CountOfTRANSMISS</th>\n",
       "      <th>AvgOfSIGMA_THETA</th>\n",
       "      <th>StDevOfSIGMA_THETA</th>\n",
       "      <th>CountOfSIGMA_THETA</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UC1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>TTCTAGCT</td>\n",
       "      <td>CACGACGTTGTAAAACGAC</td>\n",
       "      <td>GGATAACAATTTCACACAGG</td>\n",
       "      <td>N7_TTCTAGCT</td>\n",
       "      <td>N7</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>1</td>\n",
       "      <td>UC1-COI_S22_L001_R1_001.fastq.gz</td>\n",
       "      <td>...</td>\n",
       "      <td>101</td>\n",
       "      <td>4.649444</td>\n",
       "      <td>101</td>\n",
       "      <td>101</td>\n",
       "      <td>88.565879</td>\n",
       "      <td>1.576289</td>\n",
       "      <td>101</td>\n",
       "      <td>25.500296</td>\n",
       "      <td>0.326118</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC2</th>\n",
       "      <td>2.0</td>\n",
       "      <td>CCTAGAGT</td>\n",
       "      <td>CACGACGTTGTAAAACGAC</td>\n",
       "      <td>GGATAACAATTTCACACAGG</td>\n",
       "      <td>N8_CCTAGAGT</td>\n",
       "      <td>N8</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>2</td>\n",
       "      <td>UC2-COI_S23_L001_R1_001.fastq.gz</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>4.906637</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>88.278839</td>\n",
       "      <td>1.773990</td>\n",
       "      <td>100</td>\n",
       "      <td>25.359315</td>\n",
       "      <td>0.297979</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3</th>\n",
       "      <td>3.0</td>\n",
       "      <td>GCGTAAGA</td>\n",
       "      <td>CACGACGTTGTAAAACGAC</td>\n",
       "      <td>GGATAACAATTTCACACAGG</td>\n",
       "      <td>N11_GCGTAAGA</td>\n",
       "      <td>N11</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>3</td>\n",
       "      <td>UC3-COI_S24_L001_R1_001.fastq.gz</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>3.438635</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>89.038490</td>\n",
       "      <td>2.019294</td>\n",
       "      <td>100</td>\n",
       "      <td>25.820711</td>\n",
       "      <td>0.395819</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3_1</th>\n",
       "      <td>3.2</td>\n",
       "      <td>GCGTAAGA</td>\n",
       "      <td>CACGACGTTGTAAAACGAC</td>\n",
       "      <td>GGATAACAATTTCACACAGG</td>\n",
       "      <td>N9_GCGTAAGA</td>\n",
       "      <td>N9</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>19</td>\n",
       "      <td>UC3-1-COI_S40_L001_R1_001.fastq.gz</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>3.438635</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>89.038490</td>\n",
       "      <td>2.019294</td>\n",
       "      <td>100</td>\n",
       "      <td>25.820711</td>\n",
       "      <td>0.395819</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3_2</th>\n",
       "      <td>3.3</td>\n",
       "      <td>TTCTAGCT</td>\n",
       "      <td>CACGACGTTGTAAAACGAC</td>\n",
       "      <td>GGATAACAATTTCACACAGG</td>\n",
       "      <td>N10_TTCTAGCT</td>\n",
       "      <td>N10</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>17</td>\n",
       "      <td>UC3-2-COI_S38_L001_R1_001.fastq.gz</td>\n",
       "      <td>...</td>\n",
       "      <td>100</td>\n",
       "      <td>3.438635</td>\n",
       "      <td>100</td>\n",
       "      <td>100</td>\n",
       "      <td>89.038490</td>\n",
       "      <td>2.019294</td>\n",
       "      <td>100</td>\n",
       "      <td>25.820711</td>\n",
       "      <td>0.395819</td>\n",
       "      <td>100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           order tag_sequence    primer_sequence_F     primer_sequence_R  \\\n",
       "sample_ID                                                                  \n",
       "UC1          1.0     TTCTAGCT  CACGACGTTGTAAAACGAC  GGATAACAATTTCACACAGG   \n",
       "UC2          2.0     CCTAGAGT  CACGACGTTGTAAAACGAC  GGATAACAATTTCACACAGG   \n",
       "UC3          3.0     GCGTAAGA  CACGACGTTGTAAAACGAC  GGATAACAATTTCACACAGG   \n",
       "UC3_1        3.2     GCGTAAGA  CACGACGTTGTAAAACGAC  GGATAACAATTTCACACAGG   \n",
       "UC3_2        3.3     TTCTAGCT  CACGACGTTGTAAAACGAC  GGATAACAATTTCACACAGG   \n",
       "\n",
       "          library_tag_combo library    sample_type locus  tag_number  \\\n",
       "sample_ID                                                              \n",
       "UC1             N7_TTCTAGCT      N7  environmental   COI           1   \n",
       "UC2             N8_CCTAGAGT      N8  environmental   COI           2   \n",
       "UC3            N11_GCGTAAGA     N11  environmental   COI           3   \n",
       "UC3_1           N9_GCGTAAGA      N9  environmental   COI          19   \n",
       "UC3_2          N10_TTCTAGCT     N10  environmental   COI          17   \n",
       "\n",
       "                                           R1        ...         CountOfCHLA  \\\n",
       "sample_ID                                            ...                       \n",
       "UC1          UC1-COI_S22_L001_R1_001.fastq.gz        ...                 101   \n",
       "UC2          UC2-COI_S23_L001_R1_001.fastq.gz        ...                 100   \n",
       "UC3          UC3-COI_S24_L001_R1_001.fastq.gz        ...                 100   \n",
       "UC3_1      UC3-1-COI_S40_L001_R1_001.fastq.gz        ...                 100   \n",
       "UC3_2      UC3-2-COI_S38_L001_R1_001.fastq.gz        ...                 100   \n",
       "\n",
       "          AvgOfOXY_ML1 CountOfOXY_ML1 CountOfOXY_ML AvgOfTRANSMISS  \\\n",
       "sample_ID                                                            \n",
       "UC1           4.649444            101           101      88.565879   \n",
       "UC2           4.906637            100           100      88.278839   \n",
       "UC3           3.438635            100           100      89.038490   \n",
       "UC3_1         3.438635            100           100      89.038490   \n",
       "UC3_2         3.438635            100           100      89.038490   \n",
       "\n",
       "          StDevOfTRANSMISS  CountOfTRANSMISS AvgOfSIGMA_THETA  \\\n",
       "sample_ID                                                       \n",
       "UC1               1.576289               101        25.500296   \n",
       "UC2               1.773990               100        25.359315   \n",
       "UC3               2.019294               100        25.820711   \n",
       "UC3_1             2.019294               100        25.820711   \n",
       "UC3_2             2.019294               100        25.820711   \n",
       "\n",
       "           StDevOfSIGMA_THETA CountOfSIGMA_THETA  \n",
       "sample_ID                                         \n",
       "UC1                  0.326118                101  \n",
       "UC2                  0.297979                100  \n",
       "UC3                  0.395819                100  \n",
       "UC3_1                0.395819                100  \n",
       "UC3_2                0.395819                100  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = ['/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_seq_table_092519.csv',\n",
    "         '/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_otu_table_092519.csv',\n",
    "         '/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_COI_taxa_table_092519.csv']\n",
    "#dfs = [seq_COI_filt, raw_COI_filt, Ftaxa_COI_filt]\n",
    "df = pd.read_csv(files[0])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "seq_COI_filt = df.copy()\n",
    "\n",
    "df = pd.read_csv(files[1])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "raw_COI_filt = df.copy()\n",
    "\n",
    "df = pd.read_csv(files[2])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "Ftaxa_COI_filt = df.copy()\n",
    "\n",
    "raw_COI_filt.head()\n",
    "\n",
    "#import metadata\n",
    "meta_file ='/Users/kpitz/Projects/Gulf_of_California/Combined_PCTD_Metadata_043019.csv'\n",
    "df= pd.read_csv(meta_file)\n",
    "df=df.sort_values(['order'])\n",
    "df.set_index('sample_ID', inplace=True)\n",
    "print(list(df))\n",
    "\n",
    "meta_COI_PCTD = df.copy()\n",
    "meta_COI_PCTD.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OTU_1</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>Calanus</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_10</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Metridinidae</td>\n",
       "      <td>Metridia</td>\n",
       "      <td>Metridia lucens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_100</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>g_</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_100012</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Metridinidae</td>\n",
       "      <td>Metridia</td>\n",
       "      <td>Metridia lucens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_100017</th>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>Calanus</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Kingdom      Phylum         Class         Order        Family  \\\n",
       "OTU                                                                           \n",
       "OTU_1       Eukaryota  Arthropoda   Hexanauplia     Calanoida     Calanidae   \n",
       "OTU_10      Eukaryota  Arthropoda   Hexanauplia     Calanoida  Metridinidae   \n",
       "OTU_100     Eukaryota  Arthropoda  Malacostraca  Euphausiacea  Euphausiidae   \n",
       "OTU_100012  Eukaryota  Arthropoda   Hexanauplia     Calanoida  Metridinidae   \n",
       "OTU_100017  Eukaryota  Arthropoda   Hexanauplia     Calanoida     Calanidae   \n",
       "\n",
       "               Genus          Species  \n",
       "OTU                                    \n",
       "OTU_1        Calanus               s_  \n",
       "OTU_10      Metridia  Metridia lucens  \n",
       "OTU_100           g_               s_  \n",
       "OTU_100012  Metridia  Metridia lucens  \n",
       "OTU_100017   Calanus               s_  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "files = ['/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_18S_seq_table_100119.csv',\n",
    "         '/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_18S_otu_table_100119.csv',\n",
    "         '/Users/kpitz/Projects/Gulf_of_California/Decontaminated_tables/GOC_18S_taxa_table_100119.csv']\n",
    "#dfs = [seq_18S_filt, raw_18S_filt, Ftaxa_18S_filt]\n",
    "df = pd.read_csv(files[0])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "seq_18S_filt = df.copy()\n",
    "\n",
    "df = pd.read_csv(files[1])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "raw_18S_filt = df.copy()\n",
    "\n",
    "df = pd.read_csv(files[2])\n",
    "df.set_index('Unnamed: 0', inplace=True)\n",
    "df.index.rename('OTU', inplace=True)\n",
    "Ftaxa_18S_filt = df.copy()\n",
    "\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import USEARCH Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#OTU ID</th>\n",
       "      <th>CP23_1</th>\n",
       "      <th>CP23</th>\n",
       "      <th>CP23_2</th>\n",
       "      <th>GOC2a</th>\n",
       "      <th>UC10</th>\n",
       "      <th>UC1</th>\n",
       "      <th>UC12</th>\n",
       "      <th>UC13</th>\n",
       "      <th>UC14</th>\n",
       "      <th>...</th>\n",
       "      <th>UC9</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>OrganismDescriptionFromGenBank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OTU_1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>64</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>ATGCATGTCTAAGTACAGGCCTAAGCAAGGCGAAACCGCGAATGGC...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>Nematoscelis</td>\n",
       "      <td>difficilis</td>\n",
       "      <td>Nematoscelis difficilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OTU_10</td>\n",
       "      <td>21</td>\n",
       "      <td>23</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>222</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGCATGTCTAAGTATAAACTTTTATACTGTGAAACTGCGAATGGC...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Ctenophora</td>\n",
       "      <td>Tentaculata</td>\n",
       "      <td>Lobata</td>\n",
       "      <td>Ocyropsidae</td>\n",
       "      <td>Ocyropsis</td>\n",
       "      <td>inerme</td>\n",
       "      <td>Ocyropsis sp.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OTU_100</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGCATATCTAAATGCTTTGTTTTTATAACATTAGATTGCGGACGG...</td>\n",
       "      <td>Protista</td>\n",
       "      <td>Radiozoa</td>\n",
       "      <td>Polycystinea</td>\n",
       "      <td>Collodaria</td>\n",
       "      <td>Sphaerozoidae</td>\n",
       "      <td>Collozoum</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Collozoum inerme</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OTU_101</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGCATGTCTAAGTACAGGCCTAAGCAAGGCGAAACTGCGAATAGC...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>Nematoscelis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nematoscelis difficilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OTU_102</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>ATGCATGTCTAAGTACAAGCTCGTCTCGAGCGAAACTGCGGATGGC...</td>\n",
       "      <td>Animalia</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Echinoidea</td>\n",
       "      <td>Echinoida</td>\n",
       "      <td>Echinometridae</td>\n",
       "      <td>Heliocidaris</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Anthocidaris crassispina</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #OTU ID  CP23_1  CP23  CP23_2  GOC2a  UC10  UC1  UC12  UC13  UC14  \\\n",
       "0    OTU_1       1     1       0      0    58   64     0     0     1   \n",
       "1   OTU_10      21    23      35      1     0    0   222     4     0   \n",
       "2  OTU_100       1     0       0      2     0    0     0     0     0   \n",
       "3  OTU_101       0     0       0      0     0    3     0     0     0   \n",
       "4  OTU_102       0     1       0      0     0    0     0     0     0   \n",
       "\n",
       "                ...                UC9  \\\n",
       "0               ...                  1   \n",
       "1               ...                  0   \n",
       "2               ...                  0   \n",
       "3               ...                  0   \n",
       "4               ...                  0   \n",
       "\n",
       "                                            Sequence   Kingdom  \\\n",
       "0  ATGCATGTCTAAGTACAGGCCTAAGCAAGGCGAAACCGCGAATGGC...  Animalia   \n",
       "1  ATGCATGTCTAAGTATAAACTTTTATACTGTGAAACTGCGAATGGC...  Animalia   \n",
       "2  ATGCATATCTAAATGCTTTGTTTTTATAACATTAGATTGCGGACGG...  Protista   \n",
       "3  ATGCATGTCTAAGTACAGGCCTAAGCAAGGCGAAACTGCGAATAGC...  Animalia   \n",
       "4  ATGCATGTCTAAGTACAAGCTCGTCTCGAGCGAAACTGCGGATGGC...  Animalia   \n",
       "\n",
       "           Phylum          Class          Order           Family  \\\n",
       "0      Arthropoda   Malacostraca   Euphausiacea     Euphausiidae   \n",
       "1      Ctenophora    Tentaculata         Lobata      Ocyropsidae   \n",
       "2        Radiozoa   Polycystinea     Collodaria    Sphaerozoidae   \n",
       "3      Arthropoda   Malacostraca   Euphausiacea     Euphausiidae   \n",
       "4   Echinodermata     Echinoidea      Echinoida   Echinometridae   \n",
       "\n",
       "           Genus     Species  OrganismDescriptionFromGenBank  \n",
       "0   Nematoscelis  difficilis         Nematoscelis difficilis  \n",
       "1      Ocyropsis      inerme                   Ocyropsis sp.  \n",
       "2      Collozoum         NaN                Collozoum inerme  \n",
       "3   Nematoscelis         NaN         Nematoscelis difficilis  \n",
       "4   Heliocidaris         NaN        Anthocidaris crassispina  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#rarefied datasets\n",
    "Geller_18S = pd.read_csv('/Users/kpitz/Projects/Gulf_of_California/Geller_lab_data/18S_Geller_MGmodified_9Jan19.txt', sep='\t')\n",
    "Geller_18S\n",
    "#original 18S before mistake fixed (OTU_67 included and taxonomy wrong- sequences also wrong)\n",
    "#Geller_18So = pd.read_csv('/Users/kpitz/Projects/Gulf_of_California/Geller_lab_data/Incorrect_18S/18S_Geller.txt', sep='\t')\n",
    "#Geller_18So\n",
    "Geller_COI = pd.read_csv('/Users/kpitz/Projects/Gulf_of_California/Geller_lab_data/COI_Geller.txt', sep='\t')\n",
    "Geller_COI\n",
    "\n",
    "#non-rarefied datasets\n",
    "file= '/Users/kpitz/Projects/Gulf_of_California/Geller_lab_data/Not_rarefied/MBARI_18S_otu_table_10097_txt.txt'\n",
    "Geller_18S_nr = pd.read_csv(file, sep='\t')\n",
    "Geller_18S_nr\n",
    "file= '/Users/kpitz/Projects/Gulf_of_California/Geller_lab_data/Not_rarefied/MBARI_COI_otu_table_10095_txt.txt'\n",
    "Geller_COI_nr = pd.read_csv(file, sep='\t')\n",
    "Geller_18S.head()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']\n",
      "['CP23_1', 'CP23_2', 'CP23', 'GOC2a', 'GOC2b', 'UC10', 'UC12', 'UC13', 'UC14', 'UC15', 'UC1', 'UC2', 'UC3_1', 'UC3_2', 'UC3', 'UC4', 'UC5', 'UC6', 'UC7', 'UC9', 'Sequence', 'Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CP23_1</th>\n",
       "      <th>CP23</th>\n",
       "      <th>CP23_2</th>\n",
       "      <th>GOC2a</th>\n",
       "      <th>GOC2b</th>\n",
       "      <th>UC10</th>\n",
       "      <th>UC1</th>\n",
       "      <th>UC12</th>\n",
       "      <th>UC13</th>\n",
       "      <th>UC14</th>\n",
       "      <th>UC15</th>\n",
       "      <th>UC2</th>\n",
       "      <th>UC3_1</th>\n",
       "      <th>UC3</th>\n",
       "      <th>UC3_2</th>\n",
       "      <th>UC4</th>\n",
       "      <th>UC5</th>\n",
       "      <th>UC6</th>\n",
       "      <th>UC7</th>\n",
       "      <th>UC9</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#OTU ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>OTU_1</th>\n",
       "      <td>0.212719</td>\n",
       "      <td>0.245167</td>\n",
       "      <td>0.060691</td>\n",
       "      <td>9.074794</td>\n",
       "      <td>7.174748</td>\n",
       "      <td>0.189884</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.415120</td>\n",
       "      <td>1.799095</td>\n",
       "      <td>1.382671</td>\n",
       "      <td>4.411208</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_10</th>\n",
       "      <td>1.819526</td>\n",
       "      <td>1.790682</td>\n",
       "      <td>1.089432</td>\n",
       "      <td>0.116575</td>\n",
       "      <td>0.146620</td>\n",
       "      <td>2.282219</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.706046</td>\n",
       "      <td>8.487715</td>\n",
       "      <td>5.893629</td>\n",
       "      <td>14.037027</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015623</td>\n",
       "      <td>0.001803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.016224</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.243965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_100</th>\n",
       "      <td>0.183275</td>\n",
       "      <td>0.233750</td>\n",
       "      <td>0.144817</td>\n",
       "      <td>0.036655</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.182674</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007812</td>\n",
       "      <td>0.198297</td>\n",
       "      <td>0.003005</td>\n",
       "      <td>0.044467</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1000</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004807</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1001</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1002</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.194692</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1003</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.096144</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1004</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1005</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.013821</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1006</th>\n",
       "      <td>0.010215</td>\n",
       "      <td>0.015623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1007</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1008</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1009</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009014</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_101</th>\n",
       "      <td>0.116575</td>\n",
       "      <td>0.127992</td>\n",
       "      <td>0.035453</td>\n",
       "      <td>0.052879</td>\n",
       "      <td>0.172458</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.070906</td>\n",
       "      <td>0.059489</td>\n",
       "      <td>0.475913</td>\n",
       "      <td>0.094942</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1010</th>\n",
       "      <td>0.001803</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010816</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1011</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.231347</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1012</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.046870</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1013</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010215</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1014</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.027041</td>\n",
       "      <td>0.022834</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003005</td>\n",
       "      <td>0.016825</td>\n",
       "      <td>0.012018</td>\n",
       "      <td>0.022233</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1015</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1016</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.232548</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1017</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.084126</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1018</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.006610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1019</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.001803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.006610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_102</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.007211</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.006610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1020</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1021</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.007812</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.009614</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1022</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003005</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1023</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.117176</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_1024</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.136404</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_972</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.024036</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_973</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.040260</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_974</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188082</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_975</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003005</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_976</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.047471</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_977</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.074512</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_978</th>\n",
       "      <td>0.009614</td>\n",
       "      <td>0.006009</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_979</th>\n",
       "      <td>0.009614</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_98</th>\n",
       "      <td>0.132799</td>\n",
       "      <td>0.179669</td>\n",
       "      <td>0.286629</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_980</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.060691</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_981</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_982</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_983</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019229</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_984</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_985</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.058888</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_986</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.033650</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_987</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019830</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_988</th>\n",
       "      <td>0.016224</td>\n",
       "      <td>0.009614</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_989</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.005408</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_99</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.567851</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.091938</td>\n",
       "      <td>0.031848</td>\n",
       "      <td>0.113570</td>\n",
       "      <td>0.789583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_990</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.063095</td>\n",
       "      <td>0.004206</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_991</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.020431</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_992</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019830</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_993</th>\n",
       "      <td>0.067301</td>\n",
       "      <td>0.101552</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000601</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001202</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_994</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.061893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_995</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.018027</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_996</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.075713</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_997</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.056485</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_998</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008413</td>\n",
       "      <td>0.006610</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.002404</td>\n",
       "      <td>0.005408</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTU_999</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.015623</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1596 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            CP23_1      CP23    CP23_2     GOC2a     GOC2b      UC10  \\\n",
       "#OTU ID                                                                \n",
       "OTU_1     0.212719  0.245167  0.060691  9.074794  7.174748  0.189884   \n",
       "OTU_10    1.819526  1.790682  1.089432  0.116575  0.146620  2.282219   \n",
       "OTU_100   0.183275  0.233750  0.144817  0.036655  0.000000  0.182674   \n",
       "OTU_1000  0.000000  0.000000  0.000000  0.004807  0.000000  0.000000   \n",
       "OTU_1001  0.000000  0.000000  0.000000  0.000000  0.002404  0.000000   \n",
       "OTU_1002  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1003  0.000000  0.000000  0.000000  0.096144  0.000000  0.000000   \n",
       "OTU_1004  0.000000  0.000000  0.000000  0.009014  0.000000  0.000000   \n",
       "OTU_1005  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1006  0.010215  0.015623  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1007  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1008  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1009  0.000000  0.000000  0.000000  0.000000  0.000000  0.009014   \n",
       "OTU_101   0.116575  0.127992  0.035453  0.052879  0.172458  0.000601   \n",
       "OTU_1010  0.001803  0.002404  0.000000  0.010816  0.003605  0.000000   \n",
       "OTU_1011  0.000000  0.000000  0.000000  0.000000  0.000000  0.231347   \n",
       "OTU_1012  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1013  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1014  0.000000  0.000000  0.000000  0.027041  0.022834  0.000000   \n",
       "OTU_1015  0.000000  0.000000  0.000000  0.001202  0.000000  0.000000   \n",
       "OTU_1016  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1017  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1018  0.000000  0.000000  0.000000  0.001202  0.006610  0.000000   \n",
       "OTU_1019  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_102   0.000000  0.000601  0.000000  0.004206  0.007211  0.000000   \n",
       "OTU_1020  0.000000  0.000000  0.000000  0.000000  0.002404  0.000000   \n",
       "OTU_1021  0.000000  0.000000  0.000000  0.007812  0.003605  0.000601   \n",
       "OTU_1022  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1023  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1024  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "OTU_972   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_973   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_974   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_975   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_976   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_977   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_978   0.009614  0.006009  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_979   0.009614  0.001202  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_98    0.132799  0.179669  0.286629  0.000000  0.000000  0.000000   \n",
       "OTU_980   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_981   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_982   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_983   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_984   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_985   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_986   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_987   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_988   0.016224  0.009614  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_989   0.000000  0.000000  0.000000  0.000000  0.000601  0.001202   \n",
       "OTU_99    0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_990   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_991   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_992   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_993   0.067301  0.101552  0.000000  0.001202  0.000601  0.000000   \n",
       "OTU_994   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_995   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_996   0.000000  0.000000  0.000000  0.000000  0.000000  0.075713   \n",
       "OTU_997   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_998   0.000000  0.000000  0.000000  0.008413  0.006610  0.000000   \n",
       "OTU_999   0.000000  0.000000  0.000000  0.000000  0.000000  0.015623   \n",
       "\n",
       "               UC1      UC12      UC13      UC14       UC15       UC2  \\\n",
       "#OTU ID                                                                 \n",
       "OTU_1     0.000000  1.415120  1.799095  1.382671   4.411208  0.000000   \n",
       "OTU_10    0.000000  6.706046  8.487715  5.893629  14.037027  0.000000   \n",
       "OTU_100   0.000000  0.007812  0.198297  0.003005   0.044467  0.000000   \n",
       "OTU_1000  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1001  0.000000  0.000000  0.000000  0.000000   0.000601  0.000000   \n",
       "OTU_1002  0.000000  0.000000  0.000000  0.000000   0.000000  0.194692   \n",
       "OTU_1003  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1004  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1005  0.000000  0.000000  0.013821  0.000000   0.000000  0.000000   \n",
       "OTU_1006  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1007  0.000000  0.000000  0.000000  0.000000   0.000000  0.004206   \n",
       "OTU_1008  0.000000  0.009014  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1009  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_101   0.000000  0.070906  0.059489  0.475913   0.094942  0.000000   \n",
       "OTU_1010  0.000000  0.000000  0.000000  0.000000   0.003605  0.000000   \n",
       "OTU_1011  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1012  0.000000  0.000000  0.000000  0.046870   0.000000  0.000000   \n",
       "OTU_1013  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1014  0.000000  0.003005  0.016825  0.012018   0.022233  0.000000   \n",
       "OTU_1015  0.000000  0.000000  0.000000  0.006610   0.000000  0.000000   \n",
       "OTU_1016  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1017  0.000000  0.000000  0.000000  0.000000   0.000000  0.084126   \n",
       "OTU_1018  0.000000  0.000000  0.000000  0.003605   0.000000  0.000000   \n",
       "OTU_1019  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_102   0.000000  0.000000  0.000601  0.001202   0.006610  0.000000   \n",
       "OTU_1020  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1021  0.000000  0.002404  0.000000  0.009614   0.000601  0.000000   \n",
       "OTU_1022  0.001202  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_1023  0.000000  0.000000  0.000601  0.000000   0.000000  0.000000   \n",
       "OTU_1024  0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "...            ...       ...       ...       ...        ...       ...   \n",
       "OTU_972   0.000000  0.024036  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_973   0.000000  0.000000  0.000000  0.000000   0.040260  0.000000   \n",
       "OTU_974   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_975   0.000000  0.000000  0.000000  0.003005   0.000000  0.000000   \n",
       "OTU_976   0.000000  0.000000  0.000000  0.000000   0.047471  0.000000   \n",
       "OTU_977   0.000000  0.000000  0.074512  0.000000   0.000000  0.000000   \n",
       "OTU_978   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_979   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_98    0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_980   0.000000  0.060691  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_981   0.000000  0.000000  0.000000  0.004206   0.000000  0.000000   \n",
       "OTU_982   0.000000  0.004206  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_983   0.000000  0.000000  0.000000  0.000000   0.000000  0.019229   \n",
       "OTU_984   0.000000  0.000000  0.000000  0.000000   0.000000  0.000601   \n",
       "OTU_985   0.000000  0.058888  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_986   0.000000  0.033650  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_987   0.000000  0.000000  0.000000  0.019830   0.000000  0.000000   \n",
       "OTU_988   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_989   0.000000  0.000000  0.004206  0.005408   0.000000  0.000000   \n",
       "OTU_99    0.567851  0.000000  0.001202  0.000000   0.000000  0.003605   \n",
       "OTU_990   0.000000  0.000000  0.000000  0.000000   0.002404  0.000000   \n",
       "OTU_991   0.000000  0.020431  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_992   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_993   0.000000  0.000000  0.000000  0.001202   0.000000  0.000000   \n",
       "OTU_994   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_995   0.000000  0.000000  0.000000  0.000000   0.018027  0.000000   \n",
       "OTU_996   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_997   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "OTU_998   0.000000  0.000000  0.000000  0.002404   0.005408  0.000000   \n",
       "OTU_999   0.000000  0.000000  0.000000  0.000000   0.000000  0.000000   \n",
       "\n",
       "             UC3_1       UC3     UC3_2       UC4       UC5       UC6  \\\n",
       "#OTU ID                                                                \n",
       "OTU_1     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_10    0.000000  0.000000  0.015623  0.001803  0.000000  0.016224   \n",
       "OTU_100   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1001  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1002  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1003  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1004  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1005  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1006  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1007  0.000000  0.000000  0.000000  0.000000  0.000601  0.000000   \n",
       "OTU_1008  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1009  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_101   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1010  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1011  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1012  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1013  0.000000  0.000000  0.000000  0.000000  0.000000  0.010215   \n",
       "OTU_1014  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1015  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1016  0.000000  0.000000  0.000000  0.232548  0.000000  0.000000   \n",
       "OTU_1017  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1018  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1019  0.003605  0.001803  0.000000  0.000000  0.006610  0.000000   \n",
       "OTU_102   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1020  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1021  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_1022  0.000000  0.000000  0.000000  0.000000  0.003005  0.000000   \n",
       "OTU_1023  0.000000  0.000000  0.000000  0.000000  0.000000  0.117176   \n",
       "OTU_1024  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "OTU_972   0.000000  0.000000  0.000000  0.000000  0.000000  0.000601   \n",
       "OTU_973   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_974   0.000000  0.000000  0.000000  0.000000  0.000000  0.188082   \n",
       "OTU_975   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_976   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_977   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_978   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_979   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_98    0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_980   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_981   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_982   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_983   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_984   0.000000  0.000000  0.000000  0.000000  0.003605  0.000000   \n",
       "OTU_985   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_986   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_987   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_988   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_989   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_99    0.000000  0.001202  0.003605  0.004206  0.091938  0.031848   \n",
       "OTU_990   0.000000  0.000000  0.000000  0.063095  0.004206  0.001202   \n",
       "OTU_991   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_992   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_993   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_994   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_995   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_996   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_997   0.000000  0.000000  0.000000  0.056485  0.000000  0.000000   \n",
       "OTU_998   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "OTU_999   0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "               UC7       UC9  \n",
       "#OTU ID                       \n",
       "OTU_1     0.000601  0.000000  \n",
       "OTU_10    0.000000  0.243965  \n",
       "OTU_100   0.000000  0.000000  \n",
       "OTU_1000  0.000000  0.000000  \n",
       "OTU_1001  0.000000  0.000000  \n",
       "OTU_1002  0.000000  0.000000  \n",
       "OTU_1003  0.000000  0.000000  \n",
       "OTU_1004  0.000000  0.000000  \n",
       "OTU_1005  0.000000  0.000000  \n",
       "OTU_1006  0.000000  0.000000  \n",
       "OTU_1007  0.000000  0.000000  \n",
       "OTU_1008  0.000000  0.000000  \n",
       "OTU_1009  0.000000  0.000000  \n",
       "OTU_101   0.000000  0.000000  \n",
       "OTU_1010  0.000000  0.000000  \n",
       "OTU_1011  0.000000  0.000000  \n",
       "OTU_1012  0.000000  0.000000  \n",
       "OTU_1013  0.000000  0.000000  \n",
       "OTU_1014  0.000000  0.000000  \n",
       "OTU_1015  0.000000  0.000000  \n",
       "OTU_1016  0.000000  0.000000  \n",
       "OTU_1017  0.000000  0.000000  \n",
       "OTU_1018  0.000000  0.000000  \n",
       "OTU_1019  0.000000  0.000000  \n",
       "OTU_102   0.000000  0.000000  \n",
       "OTU_1020  0.000000  0.000000  \n",
       "OTU_1021  0.000000  0.000000  \n",
       "OTU_1022  0.000000  0.001202  \n",
       "OTU_1023  0.000000  0.000000  \n",
       "OTU_1024  0.136404  0.000000  \n",
       "...            ...       ...  \n",
       "OTU_972   0.000000  0.000000  \n",
       "OTU_973   0.000000  0.000000  \n",
       "OTU_974   0.000000  0.000000  \n",
       "OTU_975   0.000000  0.000000  \n",
       "OTU_976   0.000000  0.000000  \n",
       "OTU_977   0.000000  0.000000  \n",
       "OTU_978   0.000000  0.000000  \n",
       "OTU_979   0.000000  0.000000  \n",
       "OTU_98    0.000000  0.000000  \n",
       "OTU_980   0.000000  0.000000  \n",
       "OTU_981   0.000000  0.000000  \n",
       "OTU_982   0.000000  0.000000  \n",
       "OTU_983   0.000000  0.000000  \n",
       "OTU_984   0.000000  0.000000  \n",
       "OTU_985   0.000000  0.000000  \n",
       "OTU_986   0.000000  0.000000  \n",
       "OTU_987   0.000000  0.000000  \n",
       "OTU_988   0.000000  0.000000  \n",
       "OTU_989   0.000000  0.000000  \n",
       "OTU_99    0.113570  0.789583  \n",
       "OTU_990   0.001202  0.000000  \n",
       "OTU_991   0.000000  0.000000  \n",
       "OTU_992   0.019830  0.000000  \n",
       "OTU_993   0.000000  0.000000  \n",
       "OTU_994   0.000000  0.061893  \n",
       "OTU_995   0.000000  0.000000  \n",
       "OTU_996   0.000000  0.000000  \n",
       "OTU_997   0.000000  0.000000  \n",
       "OTU_998   0.000000  0.000000  \n",
       "OTU_999   0.000000  0.000000  \n",
       "\n",
       "[1596 rows x 20 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Make compositional rarefied OTU table and taxa table\n",
    "# Format Geller Data to get OTU table / taxa table\n",
    "print(list(Ftaxa_COI))\n",
    "#levels = list(Ftaxa_COI)\n",
    "levels =['Phylum', 'Class', 'Order', 'Family', 'Genus', 'Species']\n",
    "df = Geller_COI.copy()\n",
    "df['Species'] = df['Genus'] + ' ' + df['Species']\n",
    "df.fillna('no hits', inplace=True)\n",
    "df['Class']=df['Class'].str.replace('Maxillopoda', 'Hexanauplia')\n",
    "df.set_index('#OTU ID', inplace=True)\n",
    "\n",
    "df[levels]=df[levels].astype(str)\n",
    "#get rid of white space\n",
    "for i in levels:\n",
    "    df[i]=df[i].str.strip()\n",
    "\n",
    "cols = list(df)\n",
    "for i in range(len(cols)):\n",
    "    cols[i] = cols[i].replace('.','_')\n",
    "df.columns = cols\n",
    "\n",
    "Geller_COI_taxa = df[levels]\n",
    "\n",
    "Geller_COI_otu = df[['CP23_1', 'CP23', 'CP23_2', 'GOC2a', 'GOC2b', 'UC10', 'UC1', 'UC12', 'UC13', \n",
    "                     'UC14', 'UC15', 'UC2', 'UC3_1', 'UC3', 'UC3_2', 'UC4', 'UC5', 'UC6', 'UC7', 'UC9']]\n",
    "print(list(df))\n",
    "#This is rarefied data. Create a percent rarefied data OTU table\n",
    "df = Geller_COI_otu.copy()\n",
    "df=df.T\n",
    "cols = list(df)\n",
    "df['tot']=df.sum(axis=1)\n",
    "for i in cols:\n",
    "    df[i]=df[i]/df['tot'] *100\n",
    "df.drop('tot', axis=1, inplace=True)\n",
    "df=df.T\n",
    "Geller_COI_otu_comp = df.copy()\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Format Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['sample_name', 'order', 'tag_sequence', 'tag_sequence_orig', 'primer_sequence_F', 'primer_sequence_R', 'library_tag_combo', 'library', 'sample_type', 'locus', 'tag_number', 'R1', 'R2', 'Treatment', 'depth', 'dec_lat', 'dec_long', 'fluor', 'tmp', 'sal', 'sigma_theta', 'conduct', 'oxy_ml', 'transmiss', 'Description', 'site']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>order</th>\n",
       "      <th>tag_sequence</th>\n",
       "      <th>tag_sequence_orig</th>\n",
       "      <th>primer_sequence_F</th>\n",
       "      <th>primer_sequence_R</th>\n",
       "      <th>library_tag_combo</th>\n",
       "      <th>library</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>locus</th>\n",
       "      <th>tag_number</th>\n",
       "      <th>...</th>\n",
       "      <th>fluor</th>\n",
       "      <th>tmp</th>\n",
       "      <th>sal</th>\n",
       "      <th>sigma_theta</th>\n",
       "      <th>conduct</th>\n",
       "      <th>oxy_ml</th>\n",
       "      <th>transmiss</th>\n",
       "      <th>Description</th>\n",
       "      <th>site</th>\n",
       "      <th>site_Order</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sample_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UC1</th>\n",
       "      <td>1</td>\n",
       "      <td>TTCTAGCT_AA</td>\n",
       "      <td>TTCTAGCT</td>\n",
       "      <td>GGWACWGGWTGAACWGTWTAYCCYCC</td>\n",
       "      <td>TANACYTCNGGRTGNCCRAARAAYCA</td>\n",
       "      <td>N7_TTCTAGCT</td>\n",
       "      <td>N7</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.317525</td>\n",
       "      <td>12.57277917</td>\n",
       "      <td>33.31300833</td>\n",
       "      <td>25.17000417</td>\n",
       "      <td>3.879983208</td>\n",
       "      <td>5.731966667</td>\n",
       "      <td>86.42458333</td>\n",
       "      <td>PCNorth</td>\n",
       "      <td>UC1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC2</th>\n",
       "      <td>2</td>\n",
       "      <td>CCTAGAGT_AA</td>\n",
       "      <td>CCTAGAGT</td>\n",
       "      <td>GGWACWGGWTGAACWGTWTAYCCYCC</td>\n",
       "      <td>TANACYTCNGGRTGNCCRAARAAYCA</td>\n",
       "      <td>N8_CCTAGAGT</td>\n",
       "      <td>N8</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.327413043</td>\n",
       "      <td>12.72407391</td>\n",
       "      <td>33.28353913</td>\n",
       "      <td>25.11782174</td>\n",
       "      <td>3.890925304</td>\n",
       "      <td>5.685595652</td>\n",
       "      <td>85.7651913</td>\n",
       "      <td>PCNorth</td>\n",
       "      <td>UC2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3_1</th>\n",
       "      <td>19</td>\n",
       "      <td>GCGTAAGA_AT</td>\n",
       "      <td>GCGTAAGA</td>\n",
       "      <td>GGWACWGGWTGAACWGTWTAYCCYCC</td>\n",
       "      <td>TANACYTCNGGRTGNCCRAARAAYCA</td>\n",
       "      <td>N9_GCGTAAGA</td>\n",
       "      <td>N9</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>19</td>\n",
       "      <td>...</td>\n",
       "      <td>0.351265217</td>\n",
       "      <td>12.57266522</td>\n",
       "      <td>33.42426522</td>\n",
       "      <td>25.25619565</td>\n",
       "      <td>3.891625217</td>\n",
       "      <td>5.249904348</td>\n",
       "      <td>85.76833913</td>\n",
       "      <td>PCNorth</td>\n",
       "      <td>UC3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3_2</th>\n",
       "      <td>17</td>\n",
       "      <td>TTCTAGCT_AT</td>\n",
       "      <td>TTCTAGCT</td>\n",
       "      <td>GGWACWGGWTGAACWGTWTAYCCYCC</td>\n",
       "      <td>TANACYTCNGGRTGNCCRAARAAYCA</td>\n",
       "      <td>N10_TTCTAGCT</td>\n",
       "      <td>N10</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>17</td>\n",
       "      <td>...</td>\n",
       "      <td>0.351265217</td>\n",
       "      <td>12.57266522</td>\n",
       "      <td>33.42426522</td>\n",
       "      <td>25.25619565</td>\n",
       "      <td>3.891625217</td>\n",
       "      <td>5.249904348</td>\n",
       "      <td>85.76833913</td>\n",
       "      <td>PCNorth</td>\n",
       "      <td>UC3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UC3</th>\n",
       "      <td>3</td>\n",
       "      <td>GCGTAAGA_AA</td>\n",
       "      <td>GCGTAAGA</td>\n",
       "      <td>GGWACWGGWTGAACWGTWTAYCCYCC</td>\n",
       "      <td>TANACYTCNGGRTGNCCRAARAAYCA</td>\n",
       "      <td>N11_GCGTAAGA</td>\n",
       "      <td>N11</td>\n",
       "      <td>environmental</td>\n",
       "      <td>COI</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.351265217</td>\n",
       "      <td>12.57266522</td>\n",
       "      <td>33.42426522</td>\n",
       "      <td>25.25619565</td>\n",
       "      <td>3.891625217</td>\n",
       "      <td>5.249904348</td>\n",
       "      <td>85.76833913</td>\n",
       "      <td>PCNorth</td>\n",
       "      <td>UC3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             order tag_sequence tag_sequence_orig           primer_sequence_F  \\\n",
       "sample_name                                                                     \n",
       "UC1              1  TTCTAGCT_AA          TTCTAGCT  GGWACWGGWTGAACWGTWTAYCCYCC   \n",
       "UC2              2  CCTAGAGT_AA          CCTAGAGT  GGWACWGGWTGAACWGTWTAYCCYCC   \n",
       "UC3_1           19  GCGTAAGA_AT          GCGTAAGA  GGWACWGGWTGAACWGTWTAYCCYCC   \n",
       "UC3_2           17  TTCTAGCT_AT          TTCTAGCT  GGWACWGGWTGAACWGTWTAYCCYCC   \n",
       "UC3              3  GCGTAAGA_AA          GCGTAAGA  GGWACWGGWTGAACWGTWTAYCCYCC   \n",
       "\n",
       "                      primer_sequence_R library_tag_combo library  \\\n",
       "sample_name                                                         \n",
       "UC1          TANACYTCNGGRTGNCCRAARAAYCA       N7_TTCTAGCT      N7   \n",
       "UC2          TANACYTCNGGRTGNCCRAARAAYCA       N8_CCTAGAGT      N8   \n",
       "UC3_1        TANACYTCNGGRTGNCCRAARAAYCA       N9_GCGTAAGA      N9   \n",
       "UC3_2        TANACYTCNGGRTGNCCRAARAAYCA      N10_TTCTAGCT     N10   \n",
       "UC3          TANACYTCNGGRTGNCCRAARAAYCA      N11_GCGTAAGA     N11   \n",
       "\n",
       "               sample_type locus  tag_number    ...            fluor  \\\n",
       "sample_name                                     ...                    \n",
       "UC1          environmental   COI           1    ...         0.317525   \n",
       "UC2          environmental   COI           2    ...      0.327413043   \n",
       "UC3_1        environmental   COI          19    ...      0.351265217   \n",
       "UC3_2        environmental   COI          17    ...      0.351265217   \n",
       "UC3          environmental   COI           3    ...      0.351265217   \n",
       "\n",
       "                     tmp          sal  sigma_theta      conduct       oxy_ml  \\\n",
       "sample_name                                                                    \n",
       "UC1          12.57277917  33.31300833  25.17000417  3.879983208  5.731966667   \n",
       "UC2          12.72407391  33.28353913  25.11782174  3.890925304  5.685595652   \n",
       "UC3_1        12.57266522  33.42426522  25.25619565  3.891625217  5.249904348   \n",
       "UC3_2        12.57266522  33.42426522  25.25619565  3.891625217  5.249904348   \n",
       "UC3          12.57266522  33.42426522  25.25619565  3.891625217  5.249904348   \n",
       "\n",
       "               transmiss Description site site_Order  \n",
       "sample_name                                           \n",
       "UC1          86.42458333     PCNorth  UC1          1  \n",
       "UC2           85.7651913     PCNorth  UC2          2  \n",
       "UC3_1        85.76833913     PCNorth  UC3          3  \n",
       "UC3_2        85.76833913     PCNorth  UC3          3  \n",
       "UC3          85.76833913     PCNorth  UC3          3  \n",
       "\n",
       "[5 rows x 26 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta_dat = meta_COI.reset_index()\n",
    "print(list(meta_dat))\n",
    "#meta_dat['site']= meta_dat.sample_name.str.split('.').str[:-1].str.join('.')\n",
    "meta_dat['site_Order']=meta_dat['site'].str.replace('GOC2a', '16').str.replace('GOC2b', '17').str.replace('NTC', '0')\n",
    "meta_dat['site_Order']=meta_dat['site_Order'].str.extract('(\\d+)')\n",
    "meta_dat['site_Order']=meta_dat['site_Order'].astype(int)\n",
    "meta_dat= meta_dat.loc[meta_dat['sample_type']=='environmental']\n",
    "meta_dat.sort_values(['site_Order', 'sample_name'], inplace=True)\n",
    "meta_dat['sample_name']=meta_dat['sample_name'].str.replace('_COI','')\n",
    "meta_dat['sample_name']=meta_dat['sample_name'].str.replace('.','_')\n",
    "meta_dat.set_index('sample_name', inplace=True)\n",
    "\n",
    "meta_dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MEGAN imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Kingdom', 'Phylum', 'Class', 'Order', 'Family', 'Genus']"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "levels[:6]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genus</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Chloeia</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Phyllodoce &lt;polychaete&gt;</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Phragmatopoma</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Chone</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Phyllochaetopterus</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Genus  total\n",
       "0                  Chloeia      2\n",
       "1  Phyllodoce <polychaete>      2\n",
       "2            Phragmatopoma      1\n",
       "3                    Chone      1\n",
       "4       Phyllochaetopterus      3"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tax_lev = 'Genus'\n",
    "\n",
    "# Diversity layer - based on Banzai\n",
    "#Number of taxonomic annotation by Phyla / Class / Family by Region\n",
    "\n",
    "levels = list(Ftaxa_COI_filt)\n",
    "\n",
    "df=raw_COI_filt.copy()\n",
    "df=df.T\n",
    "df=pd.concat([df,meta_dat['Description']], axis=1, join='inner')\n",
    "df=df.loc[df['Description']!='0']\n",
    "print(len(df.index))\n",
    "#20samps\n",
    "df = df.groupby('Description').sum()\n",
    "df=df.astype(bool).astype(int)\n",
    "df=pd.concat([df.T, Ftaxa_COI_filt], axis=1, join='inner')\n",
    "df=df.loc[df['Kingdom']!='not assigned']\n",
    "df=df.loc[df['Kingdom']!='no hits']\n",
    "#df=df.loc[df['Class']!='c_']\n",
    "#df=df.loc[df['Class']!='unknown']\n",
    "df=df.loc[df['Genus'].isin(['g_','no hits', 'not assigned', 'unknown',''])== False]\n",
    "df=df.loc[df[tax_lev].isin(['g_','no hits', 'not assigned', 'unknown',''])== False]\n",
    "df=df.groupby(levels[:6]).sum() #now have number of genera\n",
    "#get num taxa\n",
    "df=df.astype(bool).astype(int)\n",
    "\n",
    "df.reset_index(inplace=True)\n",
    "\n",
    "#df=df.groupby(['Kingdom','Phylum','Class','Order', 'Family', 'Genus']).sum() \n",
    "df=df.groupby(levels[:6]).sum() \n",
    "#df=df.astype(bool).astype(int)\n",
    "taxa_class_COI=df.copy()\n",
    "df.reset_index(inplace=True)\n",
    "df_tot=df.copy()\n",
    "for i in ['EUNorth','EUSouth','PCNorth']:\n",
    "    df=df_tot.loc[df_tot[i]>0]\n",
    "    df=df[[tax_lev, i]]\n",
    "    #df['Score']=5\n",
    "    df.to_csv('/Users/kpitz/Projects/Graphlan_Tree/'+i+'_tree_input_genus_100319_Description.csv')\n",
    "#all\n",
    "df_tot['total'] = df_tot.sum(axis=1)\n",
    "df=df_tot.copy()\n",
    "df=df[[tax_lev, 'total']]    \n",
    "#df_tot['Score']=5\n",
    "#remove Family names after running scripts below\n",
    "#df_tot = df_tot.loc[df_tot['Genus'].isin(Fam_names)==False]\n",
    "\n",
    "#Decided to manually remove the following, might have spaces or something\n",
    "#Alviniconcha\n",
    "#Corymorphidae\n",
    "#Eudendriidae\n",
    "#Prayidae\n",
    "\n",
    "df.to_csv('/Users/kpitz/Projects/Graphlan_Tree/All_tree_input_genus_100319_Description.csv')\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Empty DataFrame\n",
      "Columns: [Total, Kingdom, Phylum, Class, Order, Family, EUNorth, EUSouth, PCNorth, total]\n",
      "Index: []\n",
      "      Total    Kingdom    Phylum       Class        Order        Family  \\\n",
      "Clio    NaN  Eukaryota  Mollusca  Gastropoda  Thecosomata  Cavoliniidae   \n",
      "\n",
      "      EUNorth  EUSouth  PCNorth  total  \n",
      "Clio        1        1        0      2  \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:12: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Total</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>EUNorth</th>\n",
       "      <th>EUSouth</th>\n",
       "      <th>PCNorth</th>\n",
       "      <th>total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Abylopsis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Abylidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acanthodoris</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>Onchidorididae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acartia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Acartiidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acrocalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Paracalanidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Agalma</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Agalmatidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Alexandrium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Gonyaulacales</td>\n",
       "      <td>Gonyaulacaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Allocentrotus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Echinoidea</td>\n",
       "      <td>Echinoida</td>\n",
       "      <td>Strongylocentrotidae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amphicaryon</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Prayidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amphipholis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Ophiuroidea</td>\n",
       "      <td>Ophiurida</td>\n",
       "      <td>Amphiuridae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amphissa</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Neogastropoda</td>\n",
       "      <td>Columbellidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amphithyrus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Platyscelidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aplysia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Aplysiidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Archiconchoecia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Ostracoda</td>\n",
       "      <td>Halocyprida</td>\n",
       "      <td>Halocyprididae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Asterionellopsis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>Fragilariophyceae</td>\n",
       "      <td>Fragilariales</td>\n",
       "      <td>Fragilariaceae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astropecten</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Asteroidea</td>\n",
       "      <td>Paxillosida</td>\n",
       "      <td>Astropectinidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Athorybia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Athorybiidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Atlanta</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Littorinimorpha</td>\n",
       "      <td>unknown</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aulopus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Aulopiformes</td>\n",
       "      <td>Aulopidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Balaenoptera</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Mammalia</td>\n",
       "      <td>Cetacea</td>\n",
       "      <td>Balaenopteridae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Balanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Sessilia</td>\n",
       "      <td>Balanidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bassia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Abylidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bathycoccus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chlorophyta</td>\n",
       "      <td>Mamiellophyceae</td>\n",
       "      <td>Mamiellales</td>\n",
       "      <td>Bathycoccaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bathylagus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Argentiniformes</td>\n",
       "      <td>Bathylagidae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Benthosema</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Myctophiformes</td>\n",
       "      <td>Myctophidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bestiolina</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Paracalanidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Botrylloides</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Ascidiacea</td>\n",
       "      <td>Stolidobranchia</td>\n",
       "      <td>Styelidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bregmaceros</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Gadiformes</td>\n",
       "      <td>Bregmacerotidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bulla</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Bullidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Calanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Callinectes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Decapoda</td>\n",
       "      <td>Portunidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sardinops</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Clupeiformes</td>\n",
       "      <td>Clupeidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scina</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Scinidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scolecithricella</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Scolecitrichidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scolecithrix</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Scolecitrichidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scomber</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Scombriformes</td>\n",
       "      <td>Scombridae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sergestes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Decapoda</td>\n",
       "      <td>Sergestidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sphaeronectes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Sphaeronectidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stephanopyxis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>Coscinodiscophyceae</td>\n",
       "      <td>Melosirales</td>\n",
       "      <td>Stephanopyxidaceae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stomias</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Stomiiformes</td>\n",
       "      <td>Stomiidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Streblospio</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Annelida</td>\n",
       "      <td>Polychaeta</td>\n",
       "      <td>Spionida</td>\n",
       "      <td>Spionidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strongylocentrotus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Echinoidea</td>\n",
       "      <td>Echinoida</td>\n",
       "      <td>Strongylocentrotidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stylasterias</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Asteroidea</td>\n",
       "      <td>Forcipulatida</td>\n",
       "      <td>Asteriidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stylocheilus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Aplysiidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Subeucalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Eucalanidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Symbiodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Suessiales</td>\n",
       "      <td>Symbiodiniaceae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Synopia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Synopiidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Temora</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Temoridae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thalassionema</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Bacillariophyta</td>\n",
       "      <td>Fragilariophyceae</td>\n",
       "      <td>Thalassionemales</td>\n",
       "      <td>Thalassionemataceae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thalia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Thaliacea</td>\n",
       "      <td>Salpida</td>\n",
       "      <td>Salpidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thliptodon</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Gymnosomata</td>\n",
       "      <td>Clionidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thysanoessa</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Trebouxia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chlorophyta</td>\n",
       "      <td>Trebouxiophyceae</td>\n",
       "      <td>Microthamniales</td>\n",
       "      <td>unknown</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triconia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Poecilostomatoida</td>\n",
       "      <td>Oncaeidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triopha</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>unknown</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triphoturus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Myctophiformes</td>\n",
       "      <td>Myctophidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tritonia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>Tritoniidae</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Undinula</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vibilia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Vibiliidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vinciguerria</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Chordata</td>\n",
       "      <td>Actinopteri</td>\n",
       "      <td>Stomiiformes</td>\n",
       "      <td>Phosichthyidae</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Woloszynskia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Lophodiniales</td>\n",
       "      <td>Lophodiniaceae</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>209 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Total    Kingdom           Phylum                Class  \\\n",
       "Abylopsis             1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Acanthodoris          1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Acartia               1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Acrocalanus           1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Agalma                1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Alexandrium           1.0  Eukaryota          unknown          Dinophyceae   \n",
       "Allocentrotus         1.0  Eukaryota    Echinodermata           Echinoidea   \n",
       "Amphicaryon           1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Amphipholis           1.0  Eukaryota    Echinodermata          Ophiuroidea   \n",
       "Amphissa              1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Amphithyrus           1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Aplysia               1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Archiconchoecia       1.0  Eukaryota       Arthropoda            Ostracoda   \n",
       "Asterionellopsis      1.0  Eukaryota  Bacillariophyta    Fragilariophyceae   \n",
       "Astropecten           1.0  Eukaryota    Echinodermata           Asteroidea   \n",
       "Athorybia             1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Atlanta               1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Aulopus               1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Balaenoptera          1.0  Eukaryota         Chordata             Mammalia   \n",
       "Balanus               1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Bassia                1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Bathycoccus           1.0  Eukaryota      Chlorophyta      Mamiellophyceae   \n",
       "Bathylagus            1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Benthosema            1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Bestiolina            1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Botrylloides          1.0  Eukaryota         Chordata           Ascidiacea   \n",
       "Bregmaceros           1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Bulla                 1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Calanus               1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Callinectes           1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "...                   ...        ...              ...                  ...   \n",
       "Sardinops             1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Scina                 1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Scolecithricella      1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Scolecithrix          1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Scomber               1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Sergestes             1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Sphaeronectes         1.0  Eukaryota         Cnidaria             Hydrozoa   \n",
       "Stephanopyxis         1.0  Eukaryota  Bacillariophyta  Coscinodiscophyceae   \n",
       "Stomias               1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Streblospio           1.0  Eukaryota         Annelida           Polychaeta   \n",
       "Strongylocentrotus    1.0  Eukaryota    Echinodermata           Echinoidea   \n",
       "Stylasterias          1.0  Eukaryota    Echinodermata           Asteroidea   \n",
       "Stylocheilus          1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Subeucalanus          1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Symbiodinium          1.0  Eukaryota          unknown          Dinophyceae   \n",
       "Synopia               1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Temora                1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Thalassionema         1.0  Eukaryota  Bacillariophyta    Fragilariophyceae   \n",
       "Thalia                1.0  Eukaryota         Chordata            Thaliacea   \n",
       "Thliptodon            1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Thysanoessa           1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Trebouxia             1.0  Eukaryota      Chlorophyta     Trebouxiophyceae   \n",
       "Triconia              1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Triopha               1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Triphoturus           1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Tritonia              1.0  Eukaryota         Mollusca           Gastropoda   \n",
       "Undinula              1.0  Eukaryota       Arthropoda          Hexanauplia   \n",
       "Vibilia               1.0  Eukaryota       Arthropoda         Malacostraca   \n",
       "Vinciguerria          1.0  Eukaryota         Chordata          Actinopteri   \n",
       "Woloszynskia          1.0  Eukaryota          unknown          Dinophyceae   \n",
       "\n",
       "                                Order                Family  EUNorth  EUSouth  \\\n",
       "Abylopsis               Siphonophorae              Abylidae        0        1   \n",
       "Acanthodoris             Nudibranchia        Onchidorididae        1        0   \n",
       "Acartia                     Calanoida            Acartiidae        1        1   \n",
       "Acrocalanus                 Calanoida         Paracalanidae        0        1   \n",
       "Agalma                  Siphonophorae           Agalmatidae        1        1   \n",
       "Alexandrium             Gonyaulacales        Gonyaulacaceae        1        1   \n",
       "Allocentrotus               Echinoida  Strongylocentrotidae        0        0   \n",
       "Amphicaryon             Siphonophorae              Prayidae        1        1   \n",
       "Amphipholis                 Ophiurida           Amphiuridae        1        0   \n",
       "Amphissa                Neogastropoda         Columbellidae        1        0   \n",
       "Amphithyrus                 Amphipoda         Platyscelidae        0        1   \n",
       "Aplysia                       unknown            Aplysiidae        1        1   \n",
       "Archiconchoecia           Halocyprida        Halocyprididae        0        1   \n",
       "Asterionellopsis        Fragilariales        Fragilariaceae        0        0   \n",
       "Astropecten               Paxillosida       Astropectinidae        1        0   \n",
       "Athorybia               Siphonophorae          Athorybiidae        0        1   \n",
       "Atlanta               Littorinimorpha               unknown        0        1   \n",
       "Aulopus                  Aulopiformes             Aulopidae        0        1   \n",
       "Balaenoptera                  Cetacea       Balaenopteridae        0        1   \n",
       "Balanus                      Sessilia             Balanidae        0        1   \n",
       "Bassia                  Siphonophorae              Abylidae        1        1   \n",
       "Bathycoccus               Mamiellales        Bathycoccaceae        1        1   \n",
       "Bathylagus            Argentiniformes          Bathylagidae        0        0   \n",
       "Benthosema             Myctophiformes           Myctophidae        0        1   \n",
       "Bestiolina                  Calanoida         Paracalanidae        0        1   \n",
       "Botrylloides          Stolidobranchia             Styelidae        0        1   \n",
       "Bregmaceros                Gadiformes       Bregmacerotidae        1        1   \n",
       "Bulla                         unknown              Bullidae        0        1   \n",
       "Calanus                     Calanoida             Calanidae        1        1   \n",
       "Callinectes                  Decapoda            Portunidae        0        1   \n",
       "...                               ...                   ...      ...      ...   \n",
       "Sardinops                Clupeiformes             Clupeidae        0        1   \n",
       "Scina                       Amphipoda              Scinidae        1        0   \n",
       "Scolecithricella            Calanoida      Scolecitrichidae        1        0   \n",
       "Scolecithrix                Calanoida      Scolecitrichidae        1        1   \n",
       "Scomber                 Scombriformes            Scombridae        0        1   \n",
       "Sergestes                    Decapoda           Sergestidae        1        0   \n",
       "Sphaeronectes           Siphonophorae       Sphaeronectidae        1        1   \n",
       "Stephanopyxis             Melosirales    Stephanopyxidaceae        0        0   \n",
       "Stomias                  Stomiiformes             Stomiidae        0        1   \n",
       "Streblospio                  Spionida             Spionidae        1        0   \n",
       "Strongylocentrotus          Echinoida  Strongylocentrotidae        1        0   \n",
       "Stylasterias            Forcipulatida            Asteriidae        1        0   \n",
       "Stylocheilus                  unknown            Aplysiidae        0        1   \n",
       "Subeucalanus                Calanoida           Eucalanidae        1        1   \n",
       "Symbiodinium               Suessiales       Symbiodiniaceae        1        1   \n",
       "Synopia                     Amphipoda            Synopiidae        0        1   \n",
       "Temora                      Calanoida             Temoridae        1        1   \n",
       "Thalassionema        Thalassionemales   Thalassionemataceae        1        1   \n",
       "Thalia                        Salpida              Salpidae        1        1   \n",
       "Thliptodon                Gymnosomata             Clionidae        0        1   \n",
       "Thysanoessa              Euphausiacea          Euphausiidae        1        1   \n",
       "Trebouxia             Microthamniales               unknown        1        0   \n",
       "Triconia            Poecilostomatoida             Oncaeidae        1        1   \n",
       "Triopha                  Nudibranchia               unknown        1        0   \n",
       "Triphoturus            Myctophiformes           Myctophidae        0        1   \n",
       "Tritonia                 Nudibranchia           Tritoniidae        1        0   \n",
       "Undinula                    Calanoida             Calanidae        0        1   \n",
       "Vibilia                     Amphipoda            Vibiliidae        1        1   \n",
       "Vinciguerria             Stomiiformes        Phosichthyidae        1        1   \n",
       "Woloszynskia            Lophodiniales        Lophodiniaceae        0        0   \n",
       "\n",
       "                    PCNorth  total  \n",
       "Abylopsis                 0      1  \n",
       "Acanthodoris              0      1  \n",
       "Acartia                   1      3  \n",
       "Acrocalanus               0      1  \n",
       "Agalma                    1      3  \n",
       "Alexandrium               1      3  \n",
       "Allocentrotus             1      1  \n",
       "Amphicaryon               1      3  \n",
       "Amphipholis               1      2  \n",
       "Amphissa                  0      1  \n",
       "Amphithyrus               0      1  \n",
       "Aplysia                   0      2  \n",
       "Archiconchoecia           0      1  \n",
       "Asterionellopsis          1      1  \n",
       "Astropecten               1      2  \n",
       "Athorybia                 1      2  \n",
       "Atlanta                   0      1  \n",
       "Aulopus                   0      1  \n",
       "Balaenoptera              0      1  \n",
       "Balanus                   1      2  \n",
       "Bassia                    0      2  \n",
       "Bathycoccus               1      3  \n",
       "Bathylagus                1      1  \n",
       "Benthosema                0      1  \n",
       "Bestiolina                0      1  \n",
       "Botrylloides              0      1  \n",
       "Bregmaceros               0      2  \n",
       "Bulla                     0      1  \n",
       "Calanus                   1      3  \n",
       "Callinectes               0      1  \n",
       "...                     ...    ...  \n",
       "Sardinops                 0      1  \n",
       "Scina                     0      1  \n",
       "Scolecithricella          1      2  \n",
       "Scolecithrix              1      3  \n",
       "Scomber                   0      1  \n",
       "Sergestes                 1      2  \n",
       "Sphaeronectes             1      3  \n",
       "Stephanopyxis             1      1  \n",
       "Stomias                   0      1  \n",
       "Streblospio               0      1  \n",
       "Strongylocentrotus        1      2  \n",
       "Stylasterias              1      2  \n",
       "Stylocheilus              0      1  \n",
       "Subeucalanus              1      3  \n",
       "Symbiodinium              1      3  \n",
       "Synopia                   0      1  \n",
       "Temora                    0      2  \n",
       "Thalassionema             0      2  \n",
       "Thalia                    1      3  \n",
       "Thliptodon                0      1  \n",
       "Thysanoessa               1      3  \n",
       "Trebouxia                 0      1  \n",
       "Triconia                  1      3  \n",
       "Triopha                   1      2  \n",
       "Triphoturus               0      1  \n",
       "Tritonia                  0      1  \n",
       "Undinula                  0      1  \n",
       "Vibilia                   0      2  \n",
       "Vinciguerria              0      2  \n",
       "Woloszynskia              1      1  \n",
       "\n",
       "[209 rows x 10 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''#Make annotation input file for Graphlan\n",
    "#Need to get annotations back out from MEGAN - sometimes don't match (USEARCH)\n",
    "#export all leaves within MEGAN (taxonomy) - csv format\n",
    "#*-ex.txt files\n",
    "file = '/Users/kpitz/Projects/Graphlan_Tree/All_tree_input_genus_100319_Description-ex.txt'\n",
    "df=pd.read_csv(file, names=['Megan','Total'])\n",
    "df.set_index('Megan', inplace=True)\n",
    "df1 = df_tot.copy()\n",
    "df1['Genus']=df1['Genus'].str.split(' ').str[0] #some megan annots had two annotations (e.g. <hydrozoan> )\n",
    "df1.set_index('Genus', inplace=True)\n",
    "#df=pd.concat([df,df_tot.set_index('Genus') ], axis=1)\n",
    "df=pd.concat([df,df1 ], axis=1)\n",
    "#problem_names\n",
    "#Bassia\n",
    "#Phyllodoce \n",
    "df2=df.loc[df['Kingdom'].isna()==True] #check for unmerged names\n",
    "print(df2)\n",
    "df2=df.loc[df['Total'].isna()==True] #check for unmerged names\n",
    "print(df2)\n",
    "df'''\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "           Name  PCNorth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Woloszynskia      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "2\n",
      "           Name  EUNorth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Lepidodinium      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "3\n",
      "           Name  EUSouth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Lepidodinium      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "Done with Analysis\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:70: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>EUNorth</th>\n",
       "      <th>EUSouth</th>\n",
       "      <th>PCNorth</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Acartia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Agalma</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Alexandrium</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Amphicaryon</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Amphipholis</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Aplysia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Astropecten</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Athorybia</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Balanus</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Bassia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Bathycoccus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Bregmaceros</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Calanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Calocalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Candacia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Centropages</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Chloeia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Clausocalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Cololabis</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Conchoecetta</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Conchoecia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Corolla</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Coscinodiscus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Cosmocalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Creseis</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Ctenocalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Cyclops</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Cylindrotheca</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Diphyes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Discoconchoecia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>Phyllodoce</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>Pleuromamma</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>Primno</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>Prorocentrum</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>Protodinium</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>Pseudevadne</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Pseudo-nitzschia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>Pseudodiaptomus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>Pseudoscourfieldia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Pycnococcus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>Pyrocystis</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>Rathbunaster</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>Rhincalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Sagitta</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>Scolecithricella</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>Scolecithrix</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>Sergestes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Sphaeronectes</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>Strongylocentrotus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>Stylasterias</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>Subeucalanus</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>Symbiodinium</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>Temora</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Thalassionema</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>Thalia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>Thysanoessa</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>Triconia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Triopha</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>Vibilia</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Vinciguerria</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Name  EUNorth  EUSouth  PCNorth  Total\n",
       "0              Acartia      1.0      1.0      1.0    3.0\n",
       "1               Agalma      1.0      1.0      1.0    3.0\n",
       "2          Alexandrium      1.0      1.0      1.0    3.0\n",
       "3          Amphicaryon      1.0      1.0      1.0    3.0\n",
       "4          Amphipholis      1.0      0.0      1.0    2.0\n",
       "5              Aplysia      1.0      1.0      0.0    2.0\n",
       "6          Astropecten      1.0      0.0      1.0    2.0\n",
       "7            Athorybia      0.0      1.0      1.0    2.0\n",
       "8              Balanus      0.0      1.0      1.0    2.0\n",
       "9               Bassia      1.0      1.0      0.0    2.0\n",
       "10         Bathycoccus      1.0      1.0      1.0    3.0\n",
       "11         Bregmaceros      1.0      1.0      0.0    2.0\n",
       "12             Calanus      1.0      1.0      1.0    3.0\n",
       "13         Calocalanus      1.0      1.0      1.0    3.0\n",
       "14            Candacia      1.0      1.0      1.0    3.0\n",
       "15         Centropages      1.0      1.0      0.0    2.0\n",
       "16             Chloeia      1.0      1.0      0.0    2.0\n",
       "17       Clausocalanus      1.0      1.0      1.0    3.0\n",
       "18           Cololabis      1.0      1.0      0.0    2.0\n",
       "19        Conchoecetta      1.0      0.0      1.0    2.0\n",
       "20          Conchoecia      1.0      1.0      1.0    3.0\n",
       "21             Corolla      1.0      1.0      0.0    2.0\n",
       "22       Coscinodiscus      1.0      0.0      1.0    2.0\n",
       "23        Cosmocalanus      1.0      1.0      0.0    2.0\n",
       "24             Creseis      1.0      1.0      1.0    3.0\n",
       "25        Ctenocalanus      1.0      1.0      1.0    3.0\n",
       "26             Cyclops      1.0      1.0      1.0    3.0\n",
       "27       Cylindrotheca      1.0      1.0      1.0    3.0\n",
       "28             Diphyes      1.0      1.0      1.0    3.0\n",
       "29     Discoconchoecia      1.0      1.0      1.0    3.0\n",
       "..                 ...      ...      ...      ...    ...\n",
       "68          Phyllodoce      1.0      0.0      1.0    2.0\n",
       "69         Pleuromamma      1.0      1.0      1.0    3.0\n",
       "70              Primno      1.0      0.0      1.0    2.0\n",
       "71        Prorocentrum      1.0      1.0      1.0    3.0\n",
       "72         Protodinium      1.0      1.0      1.0    3.0\n",
       "73         Pseudevadne      1.0      1.0      0.0    2.0\n",
       "74    Pseudo-nitzschia      1.0      0.0      1.0    2.0\n",
       "75     Pseudodiaptomus      1.0      1.0      0.0    2.0\n",
       "76  Pseudoscourfieldia      1.0      1.0      0.0    2.0\n",
       "77         Pycnococcus      1.0      1.0      1.0    3.0\n",
       "78          Pyrocystis      1.0      1.0      1.0    3.0\n",
       "79        Rathbunaster      1.0      0.0      1.0    2.0\n",
       "80         Rhincalanus      1.0      1.0      1.0    3.0\n",
       "81             Sagitta      1.0      1.0      1.0    3.0\n",
       "82    Scolecithricella      1.0      0.0      1.0    2.0\n",
       "83        Scolecithrix      1.0      1.0      1.0    3.0\n",
       "84           Sergestes      1.0      0.0      1.0    2.0\n",
       "85       Sphaeronectes      1.0      1.0      1.0    3.0\n",
       "86  Strongylocentrotus      1.0      0.0      1.0    2.0\n",
       "87        Stylasterias      1.0      0.0      1.0    2.0\n",
       "88        Subeucalanus      1.0      1.0      1.0    3.0\n",
       "89        Symbiodinium      1.0      1.0      1.0    3.0\n",
       "90              Temora      1.0      1.0      0.0    2.0\n",
       "91       Thalassionema      1.0      1.0      0.0    2.0\n",
       "92              Thalia      1.0      1.0      1.0    3.0\n",
       "93         Thysanoessa      1.0      1.0      1.0    3.0\n",
       "94            Triconia      1.0      1.0      1.0    3.0\n",
       "95             Triopha      1.0      0.0      1.0    2.0\n",
       "96             Vibilia      1.0      1.0      0.0    2.0\n",
       "97        Vinciguerria      1.0      1.0      0.0    2.0\n",
       "\n",
       "[98 rows x 5 columns]"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Need to get genus annotations back out from MEGAN\n",
    "#export all leaves within MEGAN (taxonomy) - csv format\n",
    "#*-ex.txt files\n",
    "\n",
    "#/Users/kpitz/Projects/Graphlan_Tree/Gell_18S_tree_input_genus-ex.txt\n",
    "#Want to create rings through Graphlan\n",
    "#Automate annotation file generation\n",
    "#For each annotation, mark with # genera in each class\n",
    "dfs=[]\n",
    "\n",
    "resultsFilename = '/Users/kpitz/Projects/Graphlan_Tree/GOC_Annot_Region_100719.txt'\n",
    "resultsfile = open(resultsFilename, \"w\")\n",
    "\n",
    "resultsfile.write('title')\n",
    "resultsfile.write('\\t')\n",
    "#resultsfile.write(filename.split('/')[-1].split('_')[0])\n",
    "resultsfile.write('GOC COI Region Num Genus Tree')\n",
    "resultsfile.write('\\n')\n",
    "resultsfile.write('title_font_size')\n",
    "resultsfile.write('\\t')\n",
    "resultsfile.write('10\\n')\n",
    "resultsfile.write('total_plotted_degrees\\t340\\n')\n",
    "resultsfile.write('start_rotation\\t270\\n')\n",
    "resultsfile.write('annotation_background_alpha\\t0.1')\n",
    "resultsfile.write('\\n')\n",
    "resultsfile.write('clade_marker_size\\t1\\n')\n",
    "resultsfile.write('clade_marker_edge_width\\t0.0001\\n')\n",
    "\n",
    "count=1\n",
    "names = ['Prayidae']  #caused an issue in plotting\n",
    "#for i,color in zip(['18S','COI','Gell_18S','Gell_COI'],['#0000FF','#FFA500','#FF0000','#800000']):\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    #for i,color in zip(['18S','COI','Gell_COI'],['#0000FF','#FFA500','#800000']):    \n",
    "    df=pd.read_csv('/Users/kpitz/Projects/Graphlan_Tree/'+i+'_tree_input_genus_100319_Description-ex.txt', names=['Name',i])\n",
    "    #df[i]=1\n",
    "    height='0.2'\n",
    "    dfs.append(df)\n",
    "    #For each ring\n",
    "    str_count = str(count)\n",
    "    print(str_count)\n",
    "    resultsfile.write('ring_internal_separator_thickness\\t'+str(count)+'\\t0.5\\n')\n",
    "    resultsfile.write('ring_width\\t'+str(count)+'\\t0.9\\n')\n",
    "    resultsfile.write('ring_height\\t'+str(count)+'\\t0.4\\n')\n",
    "    resultsfile.write('ring_label\\t'+str(count)+'\\t'+i+'\\n')\n",
    "    resultsfile.write('ring_label_color\\t'+str(count)+'\\t'+color+'\\n')\n",
    "    resultsfile.write('ignore_branch_len\\t'+'1'+'\\n')\n",
    "    \n",
    "    df.drop_duplicates('Name',inplace=True)\n",
    "    gen = df['Name'].tolist()\n",
    "    #gen = list(set(gen))\n",
    "    for x in gen:\n",
    "        genera =str(x)\n",
    "        if genera in names:\n",
    "            print(genera)\n",
    "            continue\n",
    "        #if genera in Fam_names:\n",
    "        #    print(genera)\n",
    "            continue\n",
    "        #if '\\t' in genera:\n",
    "        #    print(genera)\n",
    "        \n",
    "        resultsfile.write(genera+'\\tring_color\\t'+str_count+'\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tring_height\\t'+str_count+'\\t'+height+'\\n')\n",
    "    #if count >=3:\n",
    "    #    break\n",
    "    print(df.head())\n",
    "    count+=1\n",
    "\n",
    "df=pd.concat(dfs, axis=0)\n",
    "df=df.groupby('Name').sum()\n",
    "\n",
    "#label Genera that are only present in one region?\n",
    "df['Total']=df.sum(axis=1)\n",
    "df_left = df.loc[df['Total']!=1]\n",
    "df=df.loc[df['Total']==1]\n",
    "df_tot=df.copy()\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    df=df_tot.loc[df_tot[i]>0]\n",
    "    df.reset_index(inplace=True)\n",
    "    names = df['Name'].tolist()\n",
    "    for genera in names:\n",
    "        resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "        resultsfile.write(genera+'\\tclade_marker_size\\t'+'9'+'\\n')\n",
    "    \n",
    "#color grey the muliple ones\n",
    "df=df_left.reset_index()\n",
    "names = df['Name'].tolist()\n",
    "for genera in names:\n",
    "    resultsfile.write(genera+'\\tclade_marker_color\\t'+'#c6cad1'+'\\n')\n",
    "    resultsfile.write(genera+'\\tclade_marker_size\\t'+'4'+'\\n')\n",
    "\n",
    "\n",
    "#345 genera\n",
    "\n",
    "#Also want to label by Phylum? - for now don't worry about it, could merge with taxa_table\n",
    "\n",
    "#Names with issues\n",
    "#for genera in names:\n",
    "#    resultsfile.write(genera+'\\tring_color\\t'+'3'+'\\t#FF0000\\n')\n",
    "\n",
    "#[&label=\"Euphausiacea\"]  replace with Euphausiacea in nexus file\n",
    "\n",
    "resultsfile.close()\n",
    "print ('Done with Analysis')  \n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "(graphlan) mbari2030:Graphlan_Tree kpitz$ graphlan_annotate.py --annot GOC_Annot_Region_100719.txt All_tree_input_genus_100319_Description_extreme.nexus new_cutadapt_Region.xml\n",
    "(graphlan) mbari2030:Graphlan_Tree kpitz$ graphlan.py new_cutadapt_Region.xml test_Region_label.png --dpi 300 --size 6\n",
    "(graphlan) mbari2030:Graphlan_Tree kpitz$ graphlan.py new_cutadapt_Region.xml test_Region_label.svg --dpi 300 --size 6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Phylum labels in own ring\n",
    "'''\n",
    "ring_internal_separator_thickness\t4\t0.5\n",
    "ring_width\t4\t0.9\n",
    "ring_height\t4\t0.4\n",
    "ring_label\t4\tPhylum\n",
    "ring_label_color\t4\t#0000FF\n",
    "Arthropoda\tring_color\t4\tlightgrey\n",
    "Mollusca\tring_color\t4\tgrey\n",
    "Annelida\tring_color\t4\tgrey\n",
    "Actinopteri\tring_color\t4\tgrey\n",
    "Echinodermata\tring_color\t4\tgrey\n",
    "Cnidaria\tring_color\t4\tgrey\n",
    "'''\n",
    "\n",
    "\n",
    "\n",
    "#Arthropoda\tannotation\tArthropoda\n",
    "#Mollusca\tannotation\tMollusca\n",
    "#Cnidaria\tannotation\tCnidaria\n",
    "#Annelida\tannotation\tAnnelida\n",
    "#Actinopteri\tannotation\tActinopteri\n",
    "#Echinodermata\tannotation\tEchinodermata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ring_internal_separator_thickness\t4\t0.5\n",
    "ring_width\t4\t0.9\n",
    "ring_height\t4\t0.4\n",
    "ring_label\t4\tPhylum\n",
    "ring_label_color\t4\t#0000FF\n",
    "Arthropoda\tring_color\t4\tlightgrey\n",
    "Mollusca\tring_color\t4\tgrey\n",
    "Annelida\tring_color\t4\tgrey\n",
    "Actinopteri\tring_color\t4\tgrey\n",
    "Echinodermata\tring_color\t4\tgrey\n",
    "Cnidaria\tring_color\t4\tgrey\n",
    "\n",
    "Arthropoda\tannotation\tArthropoda\n",
    "Mollusca\tannotation\tMollusca\n",
    "Cnidaria\tannotation\tCnidaria\n",
    "Annelida\tannotation\tAnnelida\n",
    "Actinopteri\tannotation\tActinopteri\n",
    "Echinodermata\tannotation\tEchinodermata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add to Annot File\n",
    "'''Arthropoda\tannotation\tArthropoda\n",
    "#Mollusca\tclade_marker_color\t#1f78b4\n",
    "Mollusca\tannotation\tMollusca\n",
    "#Dinophyceae\tclade_marker_color\t#b2df8a\n",
    "Dinophyceae\tannotation\tDinophyceae\n",
    "#Cnidaria\tclade_marker_color\t#33a02c\n",
    "Cnidaria\tannotation\tCnidaria\n",
    "#Annelida\tclade_marker_color\t#fb9a99\n",
    "Annelida\tannotation\tAnnelida\n",
    "#Actinopteri\tclade_marker_color\t#e31a1c\n",
    "Actinopteri\tannotation\tActinopteri\n",
    "#Echinodermata\tclade_marker_color\t#fdbf6f\n",
    "Echinodermata\tannotation\tEchinodermata\n",
    "#Tunicata\tclade_marker_color\t#ff7f00\n",
    "Tunicata\tannotation\tTunicata\n",
    "#Stramenopiles\tclade_marker_color\t#cab2d6\n",
    "Stramenopiles\tannotation\tStramenopiles\n",
    "#Viridiplantae\tclade_marker_color\t#6a3d9a\n",
    "Viridiplantae\tannotation\tViridiplantae\n",
    "#Ctenophora\tclade_marker_color\t#ffff99\n",
    "Ctenophora\tannotation\tCtenophora\n",
    "#Chaetognatha\tclade_marker_color\t#b15928\n",
    "Chaetognatha\tannotation\tChaetognatha'''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "graphlan_annotate.py --annot GOC_Annot_Region_100319.txt All_tree_input_genus_100319_Description.nexus new_cutadapt_Region.xml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#source activate graphlan\n",
    "#in Figtree converted .tre file from MEGAN into nexus file (w/ annotations)\n",
    "#Then in order to annotate phylums, need to find and replace '\"]' with '' and \n",
    "\n",
    "graphlan.py new_cutadapt_Region.xml test_Region.png --dpi 300 --size 10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Need to get genus annotations back out from MEGAN\n",
    "#export all leaves within MEGAN (taxonomy) - csv format\n",
    "#*-ex.txt files\n",
    "\n",
    "#/Users/kpitz/Projects/Graphlan_Tree/Gell_18S_tree_input_genus-ex.txt\n",
    "#Want to create rings through Graphlan\n",
    "#Automate annotation file generation\n",
    "#For each annotation, mark with # genera in each class\n",
    "dfs=[]\n",
    "\n",
    "resultsFilename = '/Users/kpitz/Projects/Graphlan_Tree/GOC_Annot_Region_100319.txt'\n",
    "resultsfile = open(resultsFilename, \"w\")\n",
    "\n",
    "resultsfile.write('title')\n",
    "resultsfile.write('\\t')\n",
    "#resultsfile.write(filename.split('/')[-1].split('_')[0])\n",
    "resultsfile.write('GOC COI Region Num Genus Tree')\n",
    "resultsfile.write('\\n')\n",
    "resultsfile.write('title_font_size')\n",
    "resultsfile.write('\\t')\n",
    "resultsfile.write('10\\n')\n",
    "resultsfile.write('total_plotted_degrees\\t340\\n')\n",
    "resultsfile.write('start_rotation\\t270\\n')\n",
    "resultsfile.write('annotation_background_alpha\\t0.1')\n",
    "resultsfile.write('\\n')\n",
    "resultsfile.write('clade_marker_size\\t5\\n')\n",
    "resultsfile.write('clade_marker_edge_width\\t0.0001\\n')\n",
    "\n",
    "count=1\n",
    "names = ['Prayidae']  #caused an issue in plotting\n",
    "#for i,color in zip(['18S','COI','Gell_18S','Gell_COI'],['#0000FF','#FFA500','#FF0000','#800000']):\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    #for i,color in zip(['18S','COI','Gell_COI'],['#0000FF','#FFA500','#800000']):    \n",
    "    df=pd.read_csv('/Users/kpitz/Projects/Graphlan_Tree/'+i+'_tree_input_genus_100319_Description-ex.txt', names=['Name',i])\n",
    "    #df[i]=1\n",
    "    height='0.2'\n",
    "    dfs.append(df)\n",
    "    #For each ring\n",
    "    str_count = str(count)\n",
    "    print(str_count)\n",
    "    resultsfile.write('ring_internal_separator_thickness\\t'+str(count)+'\\t0.5\\n')\n",
    "    resultsfile.write('ring_width\\t'+str(count)+'\\t0.8\\n')\n",
    "    resultsfile.write('ring_height\\t'+str(count)+'\\t0.8\\n')\n",
    "    resultsfile.write('ring_label\\t'+str(count)+'\\t'+i+'\\n')\n",
    "    resultsfile.write('ring_label_color\\t'+str(count)+'\\t'+color+'\\n')\n",
    "    \n",
    "    df.drop_duplicates('Name',inplace=True)\n",
    "    gen = df['Name'].tolist()\n",
    "    #gen = list(set(gen))\n",
    "    for x in gen:\n",
    "        genera =str(x)\n",
    "        if genera in names:\n",
    "            print(genera)\n",
    "            continue\n",
    "        #if genera in Fam_names:\n",
    "        #    print(genera)\n",
    "            continue\n",
    "        #if '\\t' in genera:\n",
    "        #    print(genera)\n",
    "        \n",
    "        resultsfile.write(genera+'\\tring_color\\t'+str_count+'\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tring_height\\t'+str_count+'\\t'+height+'\\n')\n",
    "    #if count >=3:\n",
    "    #    break\n",
    "    print(df.head())\n",
    "    count+=1\n",
    "\n",
    "df=pd.concat(dfs, axis=0)\n",
    "df=df.groupby('Name').sum()\n",
    "\n",
    "#label Genera that are only present in one region?\n",
    "df['Total']=df.sum(axis=1)\n",
    "df_left = df.loc[df['Total']!=1]\n",
    "df=df.loc[df['Total']==1]\n",
    "df_tot=df.copy()\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    df=df_tot.loc[df_tot[i]>0]\n",
    "    df.reset_index(inplace=True)\n",
    "    names = df['Name'].tolist()\n",
    "    for genera in names:\n",
    "        resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "    \n",
    "#color grey the muliple ones\n",
    "df=df_left.reset_index()\n",
    "names = df['Name'].tolist()\n",
    "for genera in names:\n",
    "    resultsfile.write(genera+'\\tclade_marker_color\\t'+'#c6cad1'+'\\n')\n",
    "\n",
    "\n",
    "\n",
    "#345 genera\n",
    "\n",
    "#Also want to label by Phylum? - for now don't worry about it, could merge with taxa_table\n",
    "\n",
    "#Names with issues\n",
    "#for genera in names:\n",
    "#    resultsfile.write(genera+'\\tring_color\\t'+'3'+'\\t#FF0000\\n')\n",
    "\n",
    "#[&label=\"Euphausiacea\"]  replace with Euphausiacea in nexus file\n",
    "\n",
    "resultsfile.close()\n",
    "print ('Done with Analysis')  \n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "           Name  PCNorth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Woloszynskia      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "2\n",
      "           Name  EUNorth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Lepidodinium      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "3\n",
      "           Name  EUSouth\n",
      "0   Alexandrium      1.0\n",
      "1   Gymnodinium      1.0\n",
      "2  Lepidodinium      1.0\n",
      "3  Prorocentrum      1.0\n",
      "4    Pyrocystis      1.0\n",
      "Done with Analysis\n",
      "12\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/kpitz/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:50: FutureWarning: Sorting because non-concatenation axis is not aligned. A future version\n",
      "of pandas will change to not sort by default.\n",
      "\n",
      "To accept the future behavior, pass 'sort=False'.\n",
      "\n",
      "To retain the current behavior and silence the warning, pass 'sort=True'.\n",
      "\n",
      "/Users/kpitz/anaconda/lib/python3.5/site-packages/ipykernel/__main__.py:87: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EUNorth</th>\n",
       "      <th>EUSouth</th>\n",
       "      <th>PCNorth</th>\n",
       "      <th>Total</th>\n",
       "      <th>Kingdom</th>\n",
       "      <th>Phylum</th>\n",
       "      <th>Class</th>\n",
       "      <th>Order</th>\n",
       "      <th>Family</th>\n",
       "      <th>Species</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Chloeia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Annelida</td>\n",
       "      <td>Polychaeta</td>\n",
       "      <td>Eunicida</td>\n",
       "      <td>Amphinomidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phyllochaetopterus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Annelida</td>\n",
       "      <td>Polychaeta</td>\n",
       "      <td>Spionida</td>\n",
       "      <td>Chaetopteridae</td>\n",
       "      <td>Phyllochaetopterus prolifica</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phyllodoce</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Annelida</td>\n",
       "      <td>Polychaeta</td>\n",
       "      <td>Phyllodocida</td>\n",
       "      <td>Phyllodocidae</td>\n",
       "      <td>Phyllodoce medipapillata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Paraconchoecia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Ostracoda</td>\n",
       "      <td>Halocyprida</td>\n",
       "      <td>Halocyprididae</td>\n",
       "      <td>Paraconchoecia allotherium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Paracalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Paracalanidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Oncaea</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Poecilostomatoida</td>\n",
       "      <td>Oncaeidae</td>\n",
       "      <td>Oncaea scottodicarloi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Oithona</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Cyclopoida</td>\n",
       "      <td>Oithonidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pareucalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Eucalanidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Acartia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Acartiidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nematoscelis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mikroconchoecia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Ostracoda</td>\n",
       "      <td>Halocyprida</td>\n",
       "      <td>Halocyprididae</td>\n",
       "      <td>Mikroconchoecia sp. Os099.1.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Metridia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Metridinidae</td>\n",
       "      <td>Metridia lucens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mecynocera</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Mecynoceridae</td>\n",
       "      <td>Mecynocera clausi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vibilia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Vibiliidae</td>\n",
       "      <td>Vibilia armata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hyperioides</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Hyperiidae</td>\n",
       "      <td>Hyperioides sibaginis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Neocalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Calanidae</td>\n",
       "      <td>Neocalanus gracilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nyctiphanes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>Nyctiphanes simplex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Penilia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Branchiopoda</td>\n",
       "      <td>Diplostraca</td>\n",
       "      <td>Sididae</td>\n",
       "      <td>Penilia avirostris</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pleuromamma</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Metridinidae</td>\n",
       "      <td>Pleuromamma borealis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Primno</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Amphipoda</td>\n",
       "      <td>Phrosinidae</td>\n",
       "      <td>Primno brevidens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudevadne</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Branchiopoda</td>\n",
       "      <td>Diplostraca</td>\n",
       "      <td>Podonidae</td>\n",
       "      <td>Pseudevadne tergestina</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pseudodiaptomus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Pseudodiaptomidae</td>\n",
       "      <td>Pseudodiaptomus marinus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rhincalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Eucalanidae</td>\n",
       "      <td>Rhincalanus nasutus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scolecithricella</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Scolecitrichidae</td>\n",
       "      <td>Scolecithricella dentata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Scolecithrix</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Scolecitrichidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sergestes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Decapoda</td>\n",
       "      <td>Sergestidae</td>\n",
       "      <td>Sergestes similis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Subeucalanus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Eucalanidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Temora</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Calanoida</td>\n",
       "      <td>Temoridae</td>\n",
       "      <td>Temora discaudata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Thysanoessa</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Malacostraca</td>\n",
       "      <td>Euphausiacea</td>\n",
       "      <td>Euphausiidae</td>\n",
       "      <td>Thysanoessa gregaria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triconia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Arthropoda</td>\n",
       "      <td>Hexanauplia</td>\n",
       "      <td>Poecilostomatoida</td>\n",
       "      <td>Oncaeidae</td>\n",
       "      <td>Triconia conifera</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pandea</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Anthoathecata</td>\n",
       "      <td>Pandeidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Diphyes</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Diphyidae</td>\n",
       "      <td>Diphyes dispar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Isarachnanthus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Anthozoa</td>\n",
       "      <td>Ceriantharia</td>\n",
       "      <td>Arachnactidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Agalma</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Agalmatidae</td>\n",
       "      <td>Agalma elegans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nanomia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Cnidaria</td>\n",
       "      <td>Hydrozoa</td>\n",
       "      <td>Siphonophorae</td>\n",
       "      <td>Agalmatidae</td>\n",
       "      <td>Nanomia bijuga</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Amphipholis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Ophiuroidea</td>\n",
       "      <td>Ophiurida</td>\n",
       "      <td>Amphiuridae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Stylasterias</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Asteroidea</td>\n",
       "      <td>Forcipulatida</td>\n",
       "      <td>Asteriidae</td>\n",
       "      <td>Stylasterias forreri</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Strongylocentrotus</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Echinoidea</td>\n",
       "      <td>Echinoida</td>\n",
       "      <td>Strongylocentrotidae</td>\n",
       "      <td>Strongylocentrotus purpuratus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Ophiopteris</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Ophiuroidea</td>\n",
       "      <td>Ophiurida</td>\n",
       "      <td>Ophiocomidae</td>\n",
       "      <td>Ophiopteris papillosa</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astropecten</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Asteroidea</td>\n",
       "      <td>Paxillosida</td>\n",
       "      <td>Astropectinidae</td>\n",
       "      <td>Astropecten armatus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rathbunaster</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Echinodermata</td>\n",
       "      <td>Asteroidea</td>\n",
       "      <td>Forcipulatida</td>\n",
       "      <td>Labidiasteridae</td>\n",
       "      <td>Rathbunaster californicus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Limacina</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Thecosomata</td>\n",
       "      <td>Limacinidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Aplysia</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Aplysiidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Creseis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Thecosomata</td>\n",
       "      <td>Creseidae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Corolla</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Thecosomata</td>\n",
       "      <td>Cymbuliidae</td>\n",
       "      <td>Corolla spectabilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hermissenda</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>Glaucidae</td>\n",
       "      <td>Hermissenda opalescens</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Triopha</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Triopha catalinae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Firoloida</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Littorinimorpha</td>\n",
       "      <td>unknown</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Flabellina</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Nudibranchia</td>\n",
       "      <td>Flabellinidae</td>\n",
       "      <td>Flabellina trilineata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hyalocylis</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Mollusca</td>\n",
       "      <td>Gastropoda</td>\n",
       "      <td>Thecosomata</td>\n",
       "      <td>Cavoliniidae</td>\n",
       "      <td>Hyalocylis striata</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hubrechtella</th>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>Nemertea</td>\n",
       "      <td>Pilidiophora</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Hubrechtidae</td>\n",
       "      <td>Hubrechtella juliae</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Alexandrium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Gonyaulacales</td>\n",
       "      <td>Gonyaulacaceae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Lepidodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Gymnodiniales</td>\n",
       "      <td>Gymnodiniaceae</td>\n",
       "      <td>Lepidodinium chlorophorum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Pyrocystis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Pyrocystales</td>\n",
       "      <td>Pyrocystaceae</td>\n",
       "      <td>Pyrocystis noctiluca</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Protodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Suessiales</td>\n",
       "      <td>Suessiaceae</td>\n",
       "      <td>Protodinium simplex</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gymnodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Gymnodiniales</td>\n",
       "      <td>Gymnodiniaceae</td>\n",
       "      <td>Gymnodinium sp. NVA/RUS/2008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Phaeocystis</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Phaeocystales</td>\n",
       "      <td>Phaeocystaceae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hematodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Syndiniales</td>\n",
       "      <td>Syndiniaceae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Symbiodinium</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Suessiales</td>\n",
       "      <td>Symbiodiniaceae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Prorocentrum</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>Eukaryota</td>\n",
       "      <td>unknown</td>\n",
       "      <td>Dinophyceae</td>\n",
       "      <td>Prorocentrales</td>\n",
       "      <td>Prorocentraceae</td>\n",
       "      <td>s_</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>98 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    EUNorth  EUSouth  PCNorth  Total    Kingdom  \\\n",
       "Name                                                              \n",
       "Chloeia                 1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Phyllochaetopterus      1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Phyllodoce              1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Paraconchoecia          1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Paracalanus             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Oncaea                  1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Oithona                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Pareucalanus            1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Acartia                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Nematoscelis            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Mikroconchoecia         1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Metridia                1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Mecynocera              1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Vibilia                 1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Hyperioides             0.0      1.0      1.0    2.0  Eukaryota   \n",
       "Neocalanus              1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Nyctiphanes             1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Penilia                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Pleuromamma             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Primno                  1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Pseudevadne             1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Pseudodiaptomus         1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Rhincalanus             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Scolecithricella        1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Scolecithrix            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Sergestes               1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Subeucalanus            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Temora                  1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Thysanoessa             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Triconia                1.0      1.0      1.0    3.0  Eukaryota   \n",
       "...                     ...      ...      ...    ...        ...   \n",
       "Pandea                  1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Diphyes                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Isarachnanthus          1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Agalma                  1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Nanomia                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Amphipholis             1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Stylasterias            1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Strongylocentrotus      1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Ophiopteris             1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Astropecten             1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Rathbunaster            1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Limacina                1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Aplysia                 1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Creseis                 1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Corolla                 1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Hermissenda             1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Triopha                 1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Firoloida               1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Flabellina              1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Hyalocylis              0.0      1.0      1.0    2.0  Eukaryota   \n",
       "Hubrechtella            1.0      0.0      1.0    2.0  Eukaryota   \n",
       "Alexandrium             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Lepidodinium            1.0      1.0      0.0    2.0  Eukaryota   \n",
       "Pyrocystis              1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Protodinium             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Gymnodinium             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Phaeocystis             1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Hematodinium            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Symbiodinium            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "Prorocentrum            1.0      1.0      1.0    3.0  Eukaryota   \n",
       "\n",
       "                           Phylum         Class              Order  \\\n",
       "Name                                                                 \n",
       "Chloeia                  Annelida    Polychaeta           Eunicida   \n",
       "Phyllochaetopterus       Annelida    Polychaeta           Spionida   \n",
       "Phyllodoce               Annelida    Polychaeta       Phyllodocida   \n",
       "Paraconchoecia         Arthropoda     Ostracoda        Halocyprida   \n",
       "Paracalanus            Arthropoda   Hexanauplia          Calanoida   \n",
       "Oncaea                 Arthropoda   Hexanauplia  Poecilostomatoida   \n",
       "Oithona                Arthropoda   Hexanauplia         Cyclopoida   \n",
       "Pareucalanus           Arthropoda   Hexanauplia          Calanoida   \n",
       "Acartia                Arthropoda   Hexanauplia          Calanoida   \n",
       "Nematoscelis           Arthropoda  Malacostraca       Euphausiacea   \n",
       "Mikroconchoecia        Arthropoda     Ostracoda        Halocyprida   \n",
       "Metridia               Arthropoda   Hexanauplia          Calanoida   \n",
       "Mecynocera             Arthropoda   Hexanauplia          Calanoida   \n",
       "Vibilia                Arthropoda  Malacostraca          Amphipoda   \n",
       "Hyperioides            Arthropoda  Malacostraca          Amphipoda   \n",
       "Neocalanus             Arthropoda   Hexanauplia          Calanoida   \n",
       "Nyctiphanes            Arthropoda  Malacostraca       Euphausiacea   \n",
       "Penilia                Arthropoda  Branchiopoda        Diplostraca   \n",
       "Pleuromamma            Arthropoda   Hexanauplia          Calanoida   \n",
       "Primno                 Arthropoda  Malacostraca          Amphipoda   \n",
       "Pseudevadne            Arthropoda  Branchiopoda        Diplostraca   \n",
       "Pseudodiaptomus        Arthropoda   Hexanauplia          Calanoida   \n",
       "Rhincalanus            Arthropoda   Hexanauplia          Calanoida   \n",
       "Scolecithricella       Arthropoda   Hexanauplia          Calanoida   \n",
       "Scolecithrix           Arthropoda   Hexanauplia          Calanoida   \n",
       "Sergestes              Arthropoda  Malacostraca           Decapoda   \n",
       "Subeucalanus           Arthropoda   Hexanauplia          Calanoida   \n",
       "Temora                 Arthropoda   Hexanauplia          Calanoida   \n",
       "Thysanoessa            Arthropoda  Malacostraca       Euphausiacea   \n",
       "Triconia               Arthropoda   Hexanauplia  Poecilostomatoida   \n",
       "...                           ...           ...                ...   \n",
       "Pandea                   Cnidaria      Hydrozoa      Anthoathecata   \n",
       "Diphyes                  Cnidaria      Hydrozoa      Siphonophorae   \n",
       "Isarachnanthus           Cnidaria      Anthozoa       Ceriantharia   \n",
       "Agalma                   Cnidaria      Hydrozoa      Siphonophorae   \n",
       "Nanomia                  Cnidaria      Hydrozoa      Siphonophorae   \n",
       "Amphipholis         Echinodermata   Ophiuroidea          Ophiurida   \n",
       "Stylasterias        Echinodermata    Asteroidea      Forcipulatida   \n",
       "Strongylocentrotus  Echinodermata    Echinoidea          Echinoida   \n",
       "Ophiopteris         Echinodermata   Ophiuroidea          Ophiurida   \n",
       "Astropecten         Echinodermata    Asteroidea        Paxillosida   \n",
       "Rathbunaster        Echinodermata    Asteroidea      Forcipulatida   \n",
       "Limacina                 Mollusca    Gastropoda        Thecosomata   \n",
       "Aplysia                  Mollusca    Gastropoda            unknown   \n",
       "Creseis                  Mollusca    Gastropoda        Thecosomata   \n",
       "Corolla                  Mollusca    Gastropoda        Thecosomata   \n",
       "Hermissenda              Mollusca    Gastropoda       Nudibranchia   \n",
       "Triopha                  Mollusca    Gastropoda       Nudibranchia   \n",
       "Firoloida                Mollusca    Gastropoda    Littorinimorpha   \n",
       "Flabellina               Mollusca    Gastropoda       Nudibranchia   \n",
       "Hyalocylis               Mollusca    Gastropoda        Thecosomata   \n",
       "Hubrechtella             Nemertea  Pilidiophora            unknown   \n",
       "Alexandrium               unknown   Dinophyceae      Gonyaulacales   \n",
       "Lepidodinium              unknown   Dinophyceae      Gymnodiniales   \n",
       "Pyrocystis                unknown   Dinophyceae       Pyrocystales   \n",
       "Protodinium               unknown   Dinophyceae         Suessiales   \n",
       "Gymnodinium               unknown   Dinophyceae      Gymnodiniales   \n",
       "Phaeocystis               unknown       unknown      Phaeocystales   \n",
       "Hematodinium              unknown   Dinophyceae        Syndiniales   \n",
       "Symbiodinium              unknown   Dinophyceae         Suessiales   \n",
       "Prorocentrum              unknown   Dinophyceae     Prorocentrales   \n",
       "\n",
       "                                  Family                        Species  \n",
       "Name                                                                     \n",
       "Chloeia                     Amphinomidae                             s_  \n",
       "Phyllochaetopterus        Chaetopteridae   Phyllochaetopterus prolifica  \n",
       "Phyllodoce                 Phyllodocidae       Phyllodoce medipapillata  \n",
       "Paraconchoecia            Halocyprididae     Paraconchoecia allotherium  \n",
       "Paracalanus                Paracalanidae                             s_  \n",
       "Oncaea                         Oncaeidae          Oncaea scottodicarloi  \n",
       "Oithona                       Oithonidae                             s_  \n",
       "Pareucalanus                 Eucalanidae                             s_  \n",
       "Acartia                       Acartiidae                             s_  \n",
       "Nematoscelis                Euphausiidae                             s_  \n",
       "Mikroconchoecia           Halocyprididae  Mikroconchoecia sp. Os099.1.3  \n",
       "Metridia                    Metridinidae                Metridia lucens  \n",
       "Mecynocera                 Mecynoceridae              Mecynocera clausi  \n",
       "Vibilia                       Vibiliidae                 Vibilia armata  \n",
       "Hyperioides                   Hyperiidae          Hyperioides sibaginis  \n",
       "Neocalanus                     Calanidae            Neocalanus gracilis  \n",
       "Nyctiphanes                 Euphausiidae            Nyctiphanes simplex  \n",
       "Penilia                          Sididae             Penilia avirostris  \n",
       "Pleuromamma                 Metridinidae           Pleuromamma borealis  \n",
       "Primno                       Phrosinidae               Primno brevidens  \n",
       "Pseudevadne                    Podonidae         Pseudevadne tergestina  \n",
       "Pseudodiaptomus        Pseudodiaptomidae        Pseudodiaptomus marinus  \n",
       "Rhincalanus                  Eucalanidae            Rhincalanus nasutus  \n",
       "Scolecithricella        Scolecitrichidae       Scolecithricella dentata  \n",
       "Scolecithrix            Scolecitrichidae                             s_  \n",
       "Sergestes                    Sergestidae              Sergestes similis  \n",
       "Subeucalanus                 Eucalanidae                             s_  \n",
       "Temora                         Temoridae              Temora discaudata  \n",
       "Thysanoessa                 Euphausiidae           Thysanoessa gregaria  \n",
       "Triconia                       Oncaeidae              Triconia conifera  \n",
       "...                                  ...                            ...  \n",
       "Pandea                         Pandeidae                             s_  \n",
       "Diphyes                        Diphyidae                 Diphyes dispar  \n",
       "Isarachnanthus             Arachnactidae                             s_  \n",
       "Agalma                       Agalmatidae                 Agalma elegans  \n",
       "Nanomia                      Agalmatidae                 Nanomia bijuga  \n",
       "Amphipholis                  Amphiuridae                             s_  \n",
       "Stylasterias                  Asteriidae           Stylasterias forreri  \n",
       "Strongylocentrotus  Strongylocentrotidae  Strongylocentrotus purpuratus  \n",
       "Ophiopteris                 Ophiocomidae          Ophiopteris papillosa  \n",
       "Astropecten              Astropectinidae            Astropecten armatus  \n",
       "Rathbunaster             Labidiasteridae      Rathbunaster californicus  \n",
       "Limacina                     Limacinidae                             s_  \n",
       "Aplysia                       Aplysiidae                             s_  \n",
       "Creseis                        Creseidae                             s_  \n",
       "Corolla                      Cymbuliidae            Corolla spectabilis  \n",
       "Hermissenda                    Glaucidae         Hermissenda opalescens  \n",
       "Triopha                          unknown              Triopha catalinae  \n",
       "Firoloida                        unknown                             s_  \n",
       "Flabellina                 Flabellinidae          Flabellina trilineata  \n",
       "Hyalocylis                  Cavoliniidae             Hyalocylis striata  \n",
       "Hubrechtella                Hubrechtidae            Hubrechtella juliae  \n",
       "Alexandrium               Gonyaulacaceae                             s_  \n",
       "Lepidodinium              Gymnodiniaceae      Lepidodinium chlorophorum  \n",
       "Pyrocystis                 Pyrocystaceae           Pyrocystis noctiluca  \n",
       "Protodinium                  Suessiaceae            Protodinium simplex  \n",
       "Gymnodinium               Gymnodiniaceae   Gymnodinium sp. NVA/RUS/2008  \n",
       "Phaeocystis               Phaeocystaceae                             s_  \n",
       "Hematodinium                Syndiniaceae                             s_  \n",
       "Symbiodinium             Symbiodiniaceae                             s_  \n",
       "Prorocentrum             Prorocentraceae                             s_  \n",
       "\n",
       "[98 rows x 10 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Try making another ring alternating color by Phylum\n",
    "#Need to get genus annotations back out from MEGAN\n",
    "#export all leaves within MEGAN (taxonomy) - csv format\n",
    "#*-ex.txt files\n",
    "\n",
    "#/Users/kpitz/Projects/Graphlan_Tree/Gell_18S_tree_input_genus-ex.txt\n",
    "#Want to create rings through Graphlan\n",
    "#Automate annotation file generation\n",
    "#For each annotation, mark with # genera in each class\n",
    "dfs=[]\n",
    "\n",
    "#resultsFilename = '/Users/kpitz/Projects/Graphlan_Tree/GOC_Annot_Region_100319.txt'\n",
    "\n",
    "\n",
    "count=1\n",
    "names = ['Prayidae']  #caused an issue in plotting\n",
    "#for i,color in zip(['18S','COI','Gell_18S','Gell_COI'],['#0000FF','#FFA500','#FF0000','#800000']):\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    #for i,color in zip(['18S','COI','Gell_COI'],['#0000FF','#FFA500','#800000']):    \n",
    "    df=pd.read_csv('/Users/kpitz/Projects/Graphlan_Tree/'+i+'_tree_input_genus_100319_Description-ex.txt', names=['Name',i])\n",
    "    #df[i]=1\n",
    "    height='0.2'\n",
    "    dfs.append(df)\n",
    "    #For each ring\n",
    "    str_count = str(count)\n",
    "    print(str_count)\n",
    "\n",
    "    df.drop_duplicates('Name',inplace=True)\n",
    "    gen = df['Name'].tolist()\n",
    "    #gen = list(set(gen))\n",
    "    for x in gen:\n",
    "        genera =str(x)\n",
    "        if genera in names:\n",
    "            print(genera)\n",
    "            continue\n",
    "        #if genera in Fam_names:\n",
    "        #    print(genera)\n",
    "            continue\n",
    "        #if '\\t' in genera:\n",
    "        #    print(genera)\n",
    "        \n",
    "        #resultsfile.write(genera+'\\tring_color\\t'+str_count+'\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "        #resultsfile.write(genera+'\\tring_height\\t'+str_count+'\\t'+height+'\\n')\n",
    "    #if count >=3:\n",
    "    #    break\n",
    "    print(df.head())\n",
    "    count+=1\n",
    "\n",
    "df=pd.concat(dfs, axis=0)\n",
    "df=df.groupby('Name').sum()\n",
    "\n",
    "#label Genera that are only present in one region?\n",
    "df['Total']=df.sum(axis=1)\n",
    "df_left = df.loc[df['Total']!=1]\n",
    "df=df.loc[df['Total']==1]\n",
    "df_tot=df.copy()\n",
    "for i,color in zip(['PCNorth','EUNorth','EUSouth'],['#0000FF','#FFA500','#FF0000']):    \n",
    "    df=df_tot.loc[df_tot[i]>0]\n",
    "    df.reset_index(inplace=True)\n",
    "    names = df['Name'].tolist()\n",
    "    #for genera in names:\n",
    "        #resultsfile.write(genera+'\\tclade_marker_color\\t'+color+'\\n')\n",
    "    \n",
    "#color grey the muliple ones\n",
    "df=df_left.reset_index()\n",
    "names = df['Name'].tolist()\n",
    "#for genera in names:\n",
    "    #resultsfile.write(genera+'\\tclade_marker_color\\t'+'#c6cad1'+'\\n')\n",
    "\n",
    "\n",
    "\n",
    "#345 genera\n",
    "\n",
    "#Also want to label by Phylum? - for now don't worry about it, could merge with taxa_table\n",
    "\n",
    "#Names with issues\n",
    "#for genera in names:\n",
    "#    resultsfile.write(genera+'\\tring_color\\t'+'3'+'\\t#FF0000\\n')\n",
    "\n",
    "#[&label=\"Euphausiacea\"]  replace with Euphausiacea in nexus file\n",
    "\n",
    "#resultsfile.close()\n",
    "print ('Done with Analysis')  \n",
    "df.set_index('Name', inplace=True)\n",
    "df1=Ftaxa_COI_filt.drop_duplicates('Genus')\n",
    "df1['Genus']= df1['Genus'].str.split(' ').str[0]\n",
    "df1 = df1.set_index('Genus')\n",
    "\n",
    "df=df.join(df1)\n",
    "df=df.sort_values('Phylum')\n",
    "print('Num Unique Phyla',len(df['Phylum'].unique()))\n",
    "print(df['Phylum'].unique())\n",
    "phyla_list =[]\n",
    "#df=df.loc[df['Phylum'].isna()==True]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
